---
title: 'Introduction to bigPint plots'
author:
- name: Lindsay Rutter
date: '`r Sys.Date()`'
package: bigPint
bibliography: bigPint.bib
output:
  BiocStyle::html_document:
    toc_float: true
    tidy: TRUE
vignette: >
  \usepackage[utf8]{inputenc}
  %\VignetteIndexEntry{"Introduction"}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\VignettePackage{bigPint}
---

## In a nutshell

The `bigPint` software aims to **"Make your BIG data pint-sized"**. You can easily create modern and effective plots for your large multivariate datasets. These plots allow you to quickly examine the variability between all samples in your dataset, assess the varability between treatment groups versus between replicate groups, check for normalization issues, and discover outliers in your dataset. They also allow you to superimpose a subset of observations onto your full dataset to better understand how data subsets relates to your whole dataset. Both static and interactive plots are available for scatterplot matrices, parallel coordinate plots, litre plots, and volcano plots.

____________________________________________________________________________________

## RNA-sequencing visualization

The `bigPint` package can be useful for examining any large multivariate dataset. However, we note that the example datasets and example code in this package consider [RNA-sequencing datasets](https://en.wikipedia.org/wiki/RNA-Seq). If you are using this software for RNA-sequencing data, then it can help you confirm that the variability between your treatment groups is larger than that between your replicates and determine how various normalization techniques in popular RNA-sequencing analysis packages (such as [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) [@robinson2010edger], [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) [@love2014moderated], and [limma](https://bioconductor.org/packages/release/bioc/html/limma.html) [@ritchie2015limma]) affect your dataset. Moreover, you can easily superimpose lists of differentially expressed genes (DEGs) onto your dataset to check that they show the expected patterns (large variability between treatment groups and small variability between replicates).

____________________________________________________________________________________

## Motivation

Large multivariate datasets are common across numerous disciplinary fields. The best approach for looking at quantitative multivariate data are scatterplot matrices; parallel coordinate plots; and replicate line plots. Each of these plots enable assessing the association between multiple variables. With effective plotting tools, analysts can improve modeling; they can iterate between visualizations and modeling to enhance the models based on feedback from the visuals.

However, these plots are ineffective with large quantities of data: Overplotting can obscure important structure, and the plots can be slow to render if every observation is mapped to a graphical element. In this package, we developed more useful visualization techniques for large multivariate datasets by incorporating appropriate summaries and using interactivity.

____________________________________________________________________________________

## Additional resources

Please note that there is a [package website](https://lrutter.github.io/bigPint/index.html) that may also help users learn how to use the `bigPint` software. There, you can access vignettes, function documentation, and recent news about the package.

____________________________________________________________________________________

## Session info

Below is the output of `sessionInfo()` on the system from which this document was compiled running pandoc 1.19.2.1:

```{r, eval=TRUE, include=TRUE}
sessionInfo()
```

____________________________________________________________________________________

## References
