<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Hierarchical clustering • bigPint</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Hierarchical clustering">
<meta property="og:description" content="">
<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/createDataMetrics.html">Creating data metrics object</a>
    </li>
    <li>
      <a href="../articles/superimposeData.html">Alternatives to data metrics object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    manuscripts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/honeybee.html">Research article</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Hierarchical clustering</h1>
                        <h4 class="author">Lindsay Rutter</h4>
                        
            <h4 class="date">2019-05-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lindsayrutter/bigPint/blob/master/vignettes/clusters.Rmd"><code>vignettes/clusters.Rmd</code></a></small>
      <div class="hidden name"><code>clusters.Rmd</code></div>

    </div>

    
    
<div id="three-clustering-approaches" class="section level2">
<h2 class="hasAnchor">
<a href="#three-clustering-approaches" class="anchor"></a>Three clustering approaches</h2>
<p>The <code>bigPint</code> package incorporates a function <code><a href="../reference/plotClusters.html">plotClusters()</a></code> for users to conduct hierarchical clustering analyses on their data. This is accomplished through the <code><a href="https://www.rdocumentation.org/packages/stats/topics/hclust">hclust()</a></code> and <code><a href="https://www.rdocumentation.org/packages/stats/topics/cutree">cutree()</a></code> functions in the <a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/stats-package.html"><code>stats</code></a> package. By default, the resulting clusters will be plotted as parallel coordinate lines superimposed onto side-by-side boxplots showing the full dataset. This will allow users to quickly view the patterns in the resulting clusters. There are three main approaches to using the <code><a href="../reference/plotClusters.html">plotClusters()</a></code> function:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Approach 1:</strong> The clusters are determined by clustering only on a subset of data (such as significant genes). Only the significant genes are overlaid as parallel coordinate lines.</p></li>
<li><p><strong>Approach 2:</strong> The clusters are determined by clustering the full dataset. Then, only a subset of data (such as significant genes) are overlaid as parallel coordinate lines.</p></li>
<li><p><strong>Approach 3:</strong> The clusters are determined by clustering the full dataset. All genes are overlaid as parallel coordinate lines.</p></li>
</ol>
<p>We will now show examples for each of these three main approaches and discuss a few additional options in the <code><a href="../reference/plotClusters.html">plotClusters()</a></code> function.</p>
<hr>
</div>
<div id="approach-1" class="section level2">
<h2 class="hasAnchor">
<a href="#approach-1" class="anchor"></a>Approach 1</h2>
<p>Below, we read in the soybean iron metabolism dataset and perform a logarithm on it <span class="citation">(Lauter and Graham 2016)</span>. We determine that this dataset contains 5,604 genes. We also determine that this dataset has 113 genes that have FDR &lt; 1e-7 (n = 113). We will consider these our 113 significant genes of interest.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(grid)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(matrixStats)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(bigPint)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">"soybean_ir_sub"</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">"soybean_ir_sub_metrics"</span>)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">soybean_ir_sub[,<span class="op">-</span><span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log</a></span>(soybean_ir_sub[,<span class="op">-</span><span class="dv">1</span>]<span class="op">+</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(soybean_ir_sub)</a></code></pre></div>
<pre><code>## [1] 5604</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(soybean_ir_sub_metrics[[<span class="st">"N_P"</span>]]<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">1e-7</span>))</a></code></pre></div>
<pre><code>## [1] 113</code></pre>
<p>We then perform a hierarchical clustering of size four. By setting the <code>clusterAllData</code> parameter to a value of <code>FALSE</code>, our clustering algorithm only considers the 113 significant genes of interest. As a side note, we use the default agglomeration method “ward.D”. We then overlay these 113 genes as parallel coordinate lines across the full dataset represented as side-by-side boxplots.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">colList =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#00A600FF"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/palettes">rainbow</a></span>(<span class="dv">5</span>)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">5</span>)])</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_ir_sub, <span class="dt">dataMetrics =</span> soybean_ir_sub_metrics, <span class="dt">nC=</span><span class="dv">4</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">  <span class="dt">colList =</span> colList, <span class="dt">clusterAllData =</span> <span class="ot">FALSE</span>, <span class="dt">threshVal =</span> <span class="fl">1e-7</span>, <span class="dt">saveFile =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(ret)</a></code></pre></div>
<pre><code>## [1] "N_P_4"</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"N_P_4"</span>]])</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-2-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="approach-2" class="section level2">
<h2 class="hasAnchor">
<a href="#approach-2" class="anchor"></a>Approach 2</h2>
<p>We can perform a similar analysis as we did before, only now we will set the <code>clusterAllData</code> parameter to a value of <code>TRUE</code>. This means our four clusters will be created considering <em>all</em> genes in the data (n = 5,604). The clusters will then represent the four main patterns in the full dataset. After that, we will only overlay the 113 significant genes as parallel coordinate lines across the full dataset represented as side-by-side boxplots. Note that Cluster 1 did not contain any of the 113 significant genes.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_ir_sub, <span class="dt">dataMetrics =</span> soybean_ir_sub_metrics, <span class="dt">nC=</span><span class="dv">4</span>,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">  <span class="dt">colList =</span> colList, <span class="dt">clusterAllData =</span> <span class="ot">TRUE</span>, <span class="dt">threshVal =</span> <span class="fl">1e-7</span>, <span class="dt">saveFile =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"N_P_4"</span>]])</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-3-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="approach-3" class="section level2">
<h2 class="hasAnchor">
<a href="#approach-3" class="anchor"></a>Approach 3</h2>
<p>In this last approach, we will perform hierarchical clustering on <em>all</em> 5,604 genes in the data and also overlay <em>all</em> 5,604 genes as parallel coordinate lines. We do not create any subset of genes (such as significant genes) because we leave the <code>dataMetrics</code> object with its default value of <code>NULL</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_ir_sub, <span class="dt">nC=</span><span class="dv">4</span>, <span class="dt">colList =</span> colList,</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  <span class="dt">clusterAllData =</span> <span class="ot">TRUE</span>, <span class="dt">saveFile =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"N_P_4"</span>]])</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-4-1.png" width="700.0000032"></p>
<p>As a side note, we can use the output from Approach 3 to better understand how Approach 2 worked. Basically, in Approach 2, all 5,604 genes were used in the clustering algorithm to create four clusters. These resulted in the cluster sizes shown in Approach 3: That is, Cluster 1 had 2,334 genes; Cluster 2 had 1,997 genes; Cluster 3 had 1,992 genes; and Cluster 4 had 881 genes. However, at that point, Approach 2 only plotted the 113 significant genes. This means that none of 2,334 genes in Cluster 1 were significant; 44 of the 1,997 genes in Cluster 2 were significant; 33 of the 1,992 genes in Cluster 3 were significant; and 36 of the 881 genes in Cluster 4 were significant.</p>
<hr>
</div>
<div id="standardization" class="section level2">
<h2 class="hasAnchor">
<a href="#standardization" class="anchor"></a>Standardization</h2>
<p>When visualizing hierarchical clustering of genes, it is often recommended to consider the <em>standardized</em> values of read counts <span class="citation">(Chandrasekhar, Thangavel, and Elayaraja 2012)</span>. Below is an example of standardizing gene read counts and plotting their clusters as parallel coordinate plots superimposed onto side-by-side boxplots. The standardization transforms each gene to have a mean of zero and a standard deviation of one across its samples. This example uses the soybean cotyledon dataset, which has three treatment groups (S1, S2, and S3) <span class="citation">(Brown and Hudson 2015)</span>. Only two treatment pairs (S1 and S2; S1 and S3) return significant genes. As a result, our <code>ret</code> object will have these two comparisons.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(soybean_cn_sub)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(soybean_cn_sub_metrics)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">soybean_cn_sub_st &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(soybean_cn_sub[,<span class="op">-</span><span class="dv">1</span>]), <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  scale)))</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">soybean_cn_sub_st<span class="op">$</span>ID &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(soybean_cn_sub<span class="op">$</span>ID)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">soybean_cn_sub_st &lt;-<span class="st"> </span>soybean_cn_sub_st[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(soybean_cn_sub_st), </a>
<a class="sourceLine" id="cb10-7" data-line-number="7">  <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(soybean_cn_sub_st)<span class="op">-</span><span class="dv">1</span>)]</a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(soybean_cn_sub_st) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(soybean_cn_sub)</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">nID &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/is.finite">is.nan</a></span>(soybean_cn_sub_st[,<span class="dv">2</span>]))</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">soybean_cn_sub_st[nID,<span class="dv">2</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(soybean_cn_sub_st)] &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_cn_sub_st, <span class="dt">dataMetrics =</span> soybean_cn_sub_metrics, </a>
<a class="sourceLine" id="cb10-12" data-line-number="12">  <span class="dt">nC=</span><span class="dv">4</span>, <span class="dt">colList =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#00A600FF"</span>, <span class="st">"#CC00FFFF"</span>, <span class="st">"red"</span>, <span class="st">"darkorange"</span>), <span class="dt">lineSize =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb10-13" data-line-number="13">  <span class="dt">lineAlpha =</span> <span class="dv">1</span>, <span class="dt">clusterAllData =</span> <span class="ot">FALSE</span>, <span class="dt">aggMethod =</span> <span class="st">"average"</span>,</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">  <span class="dt">yAxisLabel =</span> <span class="st">"Standardized read count"</span>, <span class="dt">saveFile =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<pre><code>## [1] "S2_S3: There were no significant genes"</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(ret)</a></code></pre></div>
<pre><code>## [1] "S1_S2_4" "S1_S3_4" "S2_S3_4"</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"S1_S2_4"</span>]])</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"S1_S3_4"</span>]])</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-5-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="verbose-option" class="section level2">
<h2 class="hasAnchor">
<a href="#verbose-option" class="anchor"></a>Verbose option</h2>
<p>In some cases, users may wish to use the <code><a href="../reference/plotClusters.html">plotClusters()</a></code> function to create clusters and then save these clusters to use for other purposes later, such as overlaying them onto scatterplot matrices, litre plots, and volcano plots. The gene IDs of the parallel coordinate lines in each cluster can be saved as .rds files for this purpose by setting the <code>verbose</code> option to a value of <code>TRUE</code>. Below is an example of this:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_cn_sub_st, <span class="dt">dataMetrics =</span> soybean_cn_sub_metrics, <span class="dt">nC=</span><span class="dv">4</span>,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">  <span class="dt">colList =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#00A600FF"</span>, <span class="st">"#CC00FFFF"</span>, <span class="st">"red"</span>, <span class="st">"darkorange"</span>), <span class="dt">lineSize =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb15-3" data-line-number="3">  <span class="dt">lineAlpha =</span> <span class="dv">1</span>, <span class="dt">clusterAllData =</span> <span class="ot">FALSE</span>, <span class="dt">aggMethod =</span> <span class="st">"average"</span>,</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">  <span class="dt">yAxisLabel =</span> <span class="st">"Standardized read count"</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Running the command above will save numerous files into the <code>outDir</code> location. The default location is a temporary directory. You can find the exact pathway of this temporary directory by simply typing <code><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir()</a></code> into your R console. This temporary directory and the files within it remain the same until you close your R session; at that point, a new temporary directory will be created. After running the above command, you should find the following files in your temporary directory:</p>
<ul>
<li>S1_S2_4_1.rds (10 gene IDs in Cluster 1 in S1 versus S2 comparison)</li>
<li>S1_S2_4_1.jpg (Individual plot of Cluster 1 in S1 versus S2 comparison)</li>
<li>S1_S2_4_2.rds (3 gene IDs in Cluster 1 in S1 versus S2 comparison)</li>
<li>S1_S2_4_2.jpg (Individual plot of Cluster 2 in S1 versus S2 comparison)</li>
<li>S1_S2_4_3.rds (2 gene IDs in Cluster 1 in S1 versus S2 comparison)</li>
<li>S1_S2_4_3.jpg (Individual plot of Cluster 3 in S1 versus S2 comparison)</li>
<li>S1_S2_4_4.rds (1 gene IDs in Cluster 1 in S1 versus S2 comparison)</li>
<li>S1_S2_4_4.jpg (Individual plot of Cluster 4 in S1 versus S2 comparison)</li>
<li><p>S1_S2_4.jpg (Plot of all four clusters in S1 versus S2 comparison)</p></li>
<li>S1_S3_4_1.rds (19 gene IDs in Cluster 1 in S1 versus S3 comparison)</li>
<li>S1_S3_4_1.jpg (Individual plot of Cluster 1 in S1 versus S3 comparison)</li>
<li>S1_S3_4_2.rds (7 gene IDs in Cluster 2 in S1 versus S3 comparison)</li>
<li>S1_S3_4_2.jpg (Individual plot of Cluster 2 in S1 versus S3 comparison)</li>
<li>S1_S3_4_3.rds (2 gene IDs in Cluster 3 in S1 versus S3 comparison)</li>
<li>S1_S3_4_3.jpg (Individual plot of Cluster 3 in S1 versus S3 comparison)</li>
<li>S1_S3_4_4.rds (2 gene IDs in Cluster 4 in S1 versus S3 comparison)</li>
<li>S1_S3_4_4.jpg (Individual plot of Cluster 4 in S1 versus S3 comparison)</li>
<li><p>S1_S3_4.jpg (Plot of all four clusters in S1 versus S3 comparison)</p></li>
</ul>
<p>At this point, we can use any of the .rds files to examine the genes represented in the clustered parallel coordinate plots. For instance, we can read in the 19 Cluster 1 genes in the S1 versus S3 comparison as follows:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">S1S3Cluster1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(), <span class="st">"/S1_S3_4_1.rds"</span>))</a></code></pre></div>
<p>And check its contents contain the 19 gene IDs.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">S1S3Cluster1</a></code></pre></div>
<pre><code>##  [1] "Glyma18g00690.1" "Glyma08g44110.1" "Glyma01g26570.1"
##  [4] "Glyma07g09700.1" "Glyma02g40610.1" "Glyma17g17970.1"
##  [7] "Glyma19g26250.1" "Glyma10g34630.1" "Glyma14g14220.1"
## [10] "Glyma19g26710.1" "Glyma03g29150.1" "Glyma08g19245.1"
## [13] "Glyma07g01730.2" "Glyma18g25845.1" "Glyma08g22380.1"
## [16] "Glyma20g30460.1" "Glyma12g10960.1" "Glyma16g08810.1"
## [19] "Glyma18g42630.2"</code></pre>
<p>Then, we can overlay these 19 genes from Cluster 1 in the S1 versus S3 comparison as points onto the scatterplot matrix. This step can give us another perspective of these 19 genes. We may wish to use the same color (green) that we did in the parallel coordinate plots above and add an informative plot title.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotSM.html">plotSM</a></span>(<span class="dt">data =</span> soybean_cn_sub_st, <span class="dt">geneList =</span> S1S3Cluster1,</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">  <span class="dt">pointColor =</span> <span class="st">"#00A600FF"</span>, <span class="dt">saveFile =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">ret[[<span class="st">"S1_S3"</span>]] <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggplot2/topics/labs">ggtitle</a></span>(<span class="st">"Cluster 1 Genes (n=19)"</span>)</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-13-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="pre-determined-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-determined-clusters" class="anchor"></a>Pre-determined clusters</h2>
<p>Some users may perform clustering using other software and may wish to simply overlay those clusters as parallel coordinate plots using the <code>bigPint</code> package. This can be achieved in the <code><a href="../reference/plotClusters.html">plotClusters()</a></code> function using the <code>geneLists</code> variable. As an example, let us read in three clusters that each contain a character array of gene IDs. These three clusters may have been created in another software package and we simply now wish to plot them out using the <code>bigPint</code> graphics. We see that the first cluster contains 7 gene IDs and the last two clusters each contain 2 gene IDs.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">S1S3Cluster2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(), <span class="st">"/S1_S3_4_2.rds"</span>))</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">S1S3Cluster3 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(), <span class="st">"/S1_S3_4_3.rds"</span>))</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">S1S3Cluster4 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(), <span class="st">"/S1_S3_4_4.rds"</span>))</a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">S1S3Cluster2</a></code></pre></div>
<pre><code>## [1] "Glyma06g12670.1" "Glyma12g32460.1" "Glyma17g09850.1" "Glyma18g52920.1"
## [5] "Glyma01g24710.1" "Glyma04g39880.1" "Glyma05g27450.2"</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">S1S3Cluster3</a></code></pre></div>
<pre><code>## [1] "Glyma04g37510.1" "Glyma03g19880.1"</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">S1S3Cluster4</a></code></pre></div>
<pre><code>## [1] "Glyma08g11570.1" "Glyma08g19290.1"</code></pre>
<p>We can now plot these three clusters directly as parallel coordinate lines superimposed on side-by-side boxplots as follows:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">ret &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotClusters.html">plotClusters</a></span>(<span class="dt">data=</span>soybean_cn_sub_st, <span class="dt">geneLists =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(S1S3Cluster2,</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">  S1S3Cluster3, S1S3Cluster4), <span class="dt">lineAlpha =</span> <span class="dv">1</span>, <span class="dt">lineSize =</span> <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/grid/topics/grid.draw">grid.draw</a></span>(ret[[<span class="st">"S1_S3_3"</span>]])</a></code></pre></div>
<p><img src="clusters_files/figure-html/unnamed-chunk-16-1.png" width="700.0000032"></p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-brown2015developmental">
<p>Brown, Anne V., and Karen A. Hudson. 2015. “Developmental Profiling of Gene Expression in Soybean Trifoliate Leaves and Cotyledons.” <em>BMC Plant Biology</em> 15 (1). BioMed Central: 169.</p>
</div>
<div id="ref-Chandrasekhar">
<p>Chandrasekhar, T, K Thangavel, and E Elayaraja. 2012. “Effective Clustering Algorithms for Gene Expression Data.” <em>International Journal of Computer Applications</em> 1201: 4914.</p>
</div>
<div id="ref-soybeanIR">
<p>Lauter, AN Moran, and MA Graham. 2016. “NCBI Sra Bioproject Accession: PRJNA318409.”</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#three-clustering-approaches">Three clustering approaches</a></li>
      <li><a href="#approach-1">Approach 1</a></li>
      <li><a href="#approach-2">Approach 2</a></li>
      <li><a href="#approach-3">Approach 3</a></li>
      <li><a href="#standardization">Standardization</a></li>
      <li><a href="#verbose-option">Verbose option</a></li>
      <li><a href="#pre-determined-clusters">Pre-determined clusters</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/lindsayrutter">Lindsay Rutter</a>, <a href="https://github.com/dicook">Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
