<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Creating data metrics object • bigPint</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Creating data metrics object">
<meta property="og:description" content="">
<meta property="og:image" content="http://lrutter.github.io/bigPint/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/createDataMetrics.html">Creating data metrics object</a>
    </li>
    <li>
      <a href="../articles/superimposeData.html">Alternatives to data metrics object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Creating data metrics object</h1>
                        <h4 class="author">Lindsay Rutter</h4>
                        
            <h4 class="date">2018-11-17</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lrutter/bigPint/blob/master/vignettes/createDataMetrics.Rmd"><code>vignettes/createDataMetrics.Rmd</code></a></small>
      <div class="hidden name"><code>createDataMetrics.Rmd</code></div>

    </div>

    
    
<div id="data-metrics-object" class="section level2">
<h2 class="hasAnchor">
<a href="#data-metrics-object" class="anchor"></a>Data metrics object</h2>
<p>Information about the <code>dataMetrics</code> object can be found in the article <a href="https://lrutter.github.io/bigPint/articles/dataMetrics.html">Data metrics object</a>. If a researcher is using the <code>bigPint</code> package to plot RNA-seq data, then many will create the <code>dataMetrics</code> object using popular RNA-seq packages such as <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span>, and <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> <span class="citation">(Ritchie et al. 2015)</span>. These R packages will output several interesting quantitative variables for each gene in the dataset, such as log fold change, p-value, and FDR. These quantitative variables can be incorporated into the <code>dataMetrics</code> object and determine by thresholds what subset of genes will be superimposed onto plots.</p>
<hr>
</div>
<div id="example-two-treatments" class="section level2">
<h2 class="hasAnchor">
<a href="#example-two-treatments" class="anchor"></a>Example: two treatments</h2>
<p>The following example shows how to create the <code>dataMetrics</code> object called <code>soybean_ir_sub_metrics</code>, which was shown in the article <a href="https://lrutter.github.io/bigPint/articles/dataMetrics.html">Data metrics object</a> <span class="citation">(Lauter and Graham 2016)</span>. The dataset from which it is created (<code>soybean_ir_sub</code>) contains only two treatment groups, N and P. In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> package was primarily followed.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(bigPint)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">data</span>(soybean_ir_sub)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">data =<span class="st"> </span>soybean_ir_sub</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">rownames</span>(data) =<span class="st"> </span>data[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">y =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/DGEList">DGEList</a></span>(<span class="dt">counts=</span>data[,<span class="op">-</span><span class="dv">1</span>])</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">group =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">y =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/DGEList">DGEList</a></span>(<span class="dt">counts=</span>y, <span class="dt">group=</span>group)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12">Group =<span class="st"> </span><span class="kw">factor</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"N"</span>,<span class="dv">3</span>), <span class="kw">rep</span>(<span class="st">"P"</span>,<span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="dv">0</span><span class="op">+</span>Group, <span class="dt">data=</span>y<span class="op">$</span>samples)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">levels</span>(Group)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15">y &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/estimateDisp">estimateDisp</a></span>(y, design)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">fit &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/glmfit">glmFit</a></span>(y, design)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18">soybean_ir_sub_metrics &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(fit)<span class="op">-</span><span class="dv">1</span>)){</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">  <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span><span class="kw">ncol</span>(fit)){</a>
<a class="sourceLine" id="cb1-22" data-line-number="22">    contrast=<span class="kw">rep</span>(<span class="dv">0</span>,<span class="kw">ncol</span>(fit))</a>
<a class="sourceLine" id="cb1-23" data-line-number="23">    contrast[i]=<span class="dv">1</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24">    contrast[j]=<span class="op">-</span><span class="dv">1</span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25">    lrt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/glmfit">glmLRT</a></span>(fit, <span class="dt">contrast=</span>contrast)</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">    lrt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/topTags">topTags</a></span>(lrt, <span class="dt">n =</span> <span class="kw">nrow</span>(y[[<span class="dv">1</span>]]))[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb1-27" data-line-number="27">    </a>
<a class="sourceLine" id="cb1-28" data-line-number="28">    <span class="kw"><a href="http://www.rdocumentation.org/packages/data.table/topics/setDT">setDT</a></span>(lrt, <span class="dt">keep.rownames =</span> <span class="ot">TRUE</span>)[]</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">    <span class="kw">colnames</span>(lrt)[<span class="dv">1</span>] =<span class="st"> "ID"</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30">    lrt &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(lrt)</a>
<a class="sourceLine" id="cb1-31" data-line-number="31">    </a>
<a class="sourceLine" id="cb1-32" data-line-number="32">    soybean_ir_sub_metrics[[<span class="kw">paste0</span>(<span class="kw">colnames</span>(fit)[i], <span class="st">"_"</span>, <span class="kw">colnames</span>(fit)[j])]] &lt;-<span class="st"> </span>lrt</a>
<a class="sourceLine" id="cb1-33" data-line-number="33">  }</a>
<a class="sourceLine" id="cb1-34" data-line-number="34">}</a></code></pre></div>
<p>We can indeed examine the generated <code>soybean_ir_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">str</span>(soybean_ir_sub_metrics, <span class="dt">strict.width =</span> <span class="st">"wrap"</span>)</a></code></pre></div>
<pre><code>## List of 1
## $ N_P:'data.frame': 5604 obs. of 6 variables:
## ..$ ID : chr [1:5604] "Glyma.19G168700.Wm82.a2.v1"
##    "Glyma.13G293500.Wm82.a2.v1" "Glyma.05G188700.Wm82.a2.v1"
##    "Glyma.13G173100.Wm82.a2.v1" ...
## ..$ logFC : num [1:5604] -5.92 2.99 -3.51 -3.91 -3.51 ...
## ..$ logCPM: num [1:5604] 7.52 8.08 8.83 8.27 10.19 ...
## ..$ LR : num [1:5604] 266 171 167 157 154 ...
## ..$ PValue: num [1:5604] 9.18e-60 3.65e-39 2.73e-38 6.04e-36 2.58e-35 ...
## ..$ FDR : num [1:5604] 5.14e-56 1.02e-35 5.09e-35 8.46e-33 2.89e-32 ...</code></pre>
<p>And verify that it contains one list element:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">names</span>(soybean_ir_sub_metrics)</a></code></pre></div>
<pre><code>## [1] "N_P"</code></pre>
<hr>
</div>
<div id="example-three-treatments" class="section level2">
<h2 class="hasAnchor">
<a href="#example-three-treatments" class="anchor"></a>Example: three treatments</h2>
<p>The following example shows how to create the <code>dataMetrics</code> object called <code>soybean_cn_sub_metrics</code>, which was shown in the article <a href="https://lrutter.github.io/bigPint/articles/dataMetrics.html">Data metrics object</a>). The dataset from which it is created (<code>soybean_cn_sub</code>) contains three treatment groups, S1, S2, and S3 <span class="citation">(Brown and Hudson 2015)</span>. In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> package was primarily followed.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw">data</span>(soybean_cn_sub)</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">data =<span class="st"> </span>soybean_cn_sub</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="kw">rownames</span>(data) =<span class="st"> </span>data[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb6-7" data-line-number="7"></a>
<a class="sourceLine" id="cb6-8" data-line-number="8">y =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/DGEList">DGEList</a></span>(<span class="dt">counts=</span>data[,<span class="op">-</span><span class="dv">1</span>])</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">group =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>)</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">y =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/DGEList">DGEList</a></span>(<span class="dt">counts=</span>y, <span class="dt">group=</span>group)</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">Group =<span class="st"> </span><span class="kw">factor</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"S1"</span>,<span class="dv">3</span>), <span class="kw">rep</span>(<span class="st">"S2"</span>,<span class="dv">3</span>), <span class="kw">rep</span>(<span class="st">"S3"</span>,<span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb6-12" data-line-number="12">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="dv">0</span><span class="op">+</span>Group, <span class="dt">data=</span>y<span class="op">$</span>samples)</a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">levels</span>(Group)</a>
<a class="sourceLine" id="cb6-14" data-line-number="14">y &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/estimateDisp">estimateDisp</a></span>(y, design)</a>
<a class="sourceLine" id="cb6-15" data-line-number="15">fit &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/glmfit">glmFit</a></span>(y, design)</a>
<a class="sourceLine" id="cb6-16" data-line-number="16"></a>
<a class="sourceLine" id="cb6-17" data-line-number="17">soybean_cn_sub_metrics &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb6-18" data-line-number="18"></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(fit)<span class="op">-</span><span class="dv">1</span>)){</a>
<a class="sourceLine" id="cb6-20" data-line-number="20">  <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span><span class="kw">ncol</span>(fit)){</a>
<a class="sourceLine" id="cb6-21" data-line-number="21">    contrast=<span class="kw">rep</span>(<span class="dv">0</span>,<span class="kw">ncol</span>(fit))</a>
<a class="sourceLine" id="cb6-22" data-line-number="22">    contrast[i]=<span class="dv">1</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23">    contrast[j]=<span class="op">-</span><span class="dv">1</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24">    lrt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/glmfit">glmLRT</a></span>(fit, <span class="dt">contrast=</span>contrast)</a>
<a class="sourceLine" id="cb6-25" data-line-number="25">    lrt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/edgeR/topics/topTags">topTags</a></span>(lrt, <span class="dt">n =</span> <span class="kw">nrow</span>(y[[<span class="dv">1</span>]]))[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb6-26" data-line-number="26">    </a>
<a class="sourceLine" id="cb6-27" data-line-number="27">    <span class="kw"><a href="http://www.rdocumentation.org/packages/data.table/topics/setDT">setDT</a></span>(lrt, <span class="dt">keep.rownames =</span> <span class="ot">TRUE</span>)[]</a>
<a class="sourceLine" id="cb6-28" data-line-number="28">    <span class="kw">colnames</span>(lrt)[<span class="dv">1</span>] =<span class="st"> "ID"</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29">    lrt &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(lrt)</a>
<a class="sourceLine" id="cb6-30" data-line-number="30">    </a>
<a class="sourceLine" id="cb6-31" data-line-number="31">    soybean_cn_sub_metrics[[<span class="kw">paste0</span>(<span class="kw">colnames</span>(fit)[i], <span class="st">"_"</span>, <span class="kw">colnames</span>(fit)[j])]] &lt;-<span class="st"> </span>lrt</a>
<a class="sourceLine" id="cb6-32" data-line-number="32">  }</a>
<a class="sourceLine" id="cb6-33" data-line-number="33">}</a></code></pre></div>
<p>We can indeed examine the generated <code>soybean_cn_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">str</span>(soybean_cn_sub_metrics, <span class="dt">strict.width =</span> <span class="st">"wrap"</span>)</a></code></pre></div>
<pre><code>## List of 3
## $ S1_S2:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma18g00690.1" "Glyma08g22380.1"
##    "Glyma20g30460.1" "Glyma07g09700.1" ...
## ..$ logFC : num [1:7332] 3.14 2.62 2.5 2.37 2.74 ...
## ..$ logCPM: num [1:7332] 8.04 8.05 8.14 8.19 7.93 ...
## ..$ LR : num [1:7332] 29.3 24.8 24 22.7 21 ...
## ..$ PValue: num [1:7332] 6.08e-08 6.50e-07 9.82e-07 1.94e-06 4.57e-06 ...
## ..$ FDR : num [1:7332] 0.000446 0.002384 0.0024 0.003557 0.006697 ...
## $ S1_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g22380.1" "Glyma08g19290.1"
##    "Glyma20g30460.1" "Glyma18g00690.1" ...
## ..$ logFC : num [1:7332] 3.18 3.32 2.44 2.46 3.03 ...
## ..$ logCPM: num [1:7332] 8.05 8.14 8.14 8.04 7.85 ...
## ..$ LR : num [1:7332] 30.4 24.7 23.3 22.5 22.3 ...
## ..$ PValue: num [1:7332] 3.60e-08 6.53e-07 1.42e-06 2.09e-06 2.30e-06 ...
## ..$ FDR : num [1:7332] 0.000264 0.002393 0.003378 0.003378 0.003378 ...
## $ S2_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g14670.3" "Glyma08g14670.2"
##    "Glyma08g19290.1" "Glyma06g46701.1" ...
## ..$ logFC : num [1:7332] 2.66 2.72 2.78 2.19 2.29 ...
## ..$ logCPM: num [1:7332] 7.86 7.82 8.14 7.63 7.84 ...
## ..$ LR : num [1:7332] 16.1 16 14.3 11.8 10.5 ...
## ..$ PValue: num [1:7332] 6.01e-05 6.30e-05 1.57e-04 5.81e-04 1.21e-03 ...
## ..$ FDR : num [1:7332] 0.231 0.231 0.385 1 1 ...</code></pre>
<p>And verify that it contains three list element:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">names</span>(soybean_cn_sub_metrics)</a></code></pre></div>
<pre><code>## [1] "S1_S2" "S1_S3" "S2_S3"</code></pre>
<hr>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-brown2015developmental">
<p>Brown, Anne V., and Karen A. Hudson. 2015. “Developmental Profiling of Gene Expression in Soybean Trifoliate Leaves and Cotyledons.” <em>BMC Plant Biology</em> 15 (1). BioMed Central: 169.</p>
</div>
<div id="ref-soybeanIR">
<p>Lauter, AN Moran, and MA Graham. 2016. “NCBI Sra Bioproject Accession: PRJNA318409.”</p>
</div>
<div id="ref-love2014moderated">
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12). BioMed Central: 550.</p>
</div>
<div id="ref-ritchie2015limma">
<p>Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. “Limma Powers Differential Expression Analyses for Rna-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7). Oxford University Press: e47–e47.</p>
</div>
<div id="ref-robinson2010edger">
<p>Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. “EdgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.” <em>Bioinformatics</em> 26 (1). Oxford University Press: 139–40.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#data-metrics-object">Data metrics object</a></li>
      <li><a href="#example-two-treatments">Example: two treatments</a></li>
      <li><a href="#example-three-treatments">Example: three treatments</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/lrutter">Lindsay Rutter</a>, <a href="https://github.com/dicook">Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  
<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script><script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
