<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data metrics object • bigPint</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Data metrics object">
<meta property="og:description" content="bigPint">
<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/summarizedExperiment.html">SummarizedExperiment object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/manuscripts.html">
    <span class="fa fa-pencil-square-o"></span>
     
    manuscripts
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data metrics object</h1>
                        <h4 class="author">Lindsay Rutter</h4>
                        
            <h4 class="date">2020-08-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lindsayrutter/bigPint/blob/master/vignettes/dataMetrics.Rmd"><code>vignettes/dataMetrics.Rmd</code></a></small>
      <div class="hidden name"><code>dataMetrics.Rmd</code></div>

    </div>

    
    
<div id="about-data-metrics-object" class="section level2">
<h2 class="hasAnchor">
<a href="#about-data-metrics-object" class="anchor"></a>About data metrics object</h2>
<p>Researchers may wish to superimpose a subset of the full dataset onto the full dataset. If a researcher is using the package to visualize RNA-seq data, then this subset of data is often differentially expressed genes (DEGs) returned from a model. In this case, the user may wish to use the <code>dataMetrics</code> input parameter, which contains at least one quantitative variable returned from a model such as FDR, p-value, and log fold change.</p>
<hr>
</div>
<div id="example-two-groups" class="section level2">
<h2 class="hasAnchor">
<a href="#example-two-groups" class="anchor"></a>Example: two groups</h2>
<p>As was shown in the article <a href="https://lindsayrutter.github.io/bigPint/articles/data.html">Data object</a>, the <code>data</code> object called <code>soybean_ir_sub</code> contained 5,604 genes and two treatment groups, N and P <span class="citation">(Lauter and Graham 2016)</span>. We can examine the structure of its corresponding <code>dataMetrics</code> object called <code>soybean_ir_sub_metrics</code> as follows:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">bigPint</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"soybean_ir_sub_metrics"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_ir_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 1
## $ N_P:'data.frame': 5604 obs. of 6 variables:
## ..$ ID : chr [1:5604] "Glyma.19G168700.Wm82.a2.v1" "Glyma.13G293500.Wm82.a2.v1"
##    "Glyma.05G188700.Wm82.a2.v1" "Glyma.13G173100.Wm82.a2.v1" ...
## ..$ logFC : num [1:5604] -5.92 2.99 -3.51 -3.91 -3.51 ...
## ..$ logCPM: num [1:5604] 7.52 8.08 8.83 8.27 10.19 ...
## ..$ LR : num [1:5604] 266 171 167 157 154 ...
## ..$ PValue: num [1:5604] 9.18e-60 3.65e-39 2.73e-38 6.04e-36 2.58e-35 ...
## ..$ FDR : num [1:5604] 5.14e-56 1.02e-35 5.09e-35 8.46e-33 2.89e-32 ...</code></pre>
<hr>
</div>
<div id="example-three-groups" class="section level2">
<h2 class="hasAnchor">
<a href="#example-three-groups" class="anchor"></a>Example: three groups</h2>
<p>Similarly, as was shown in the data page, the <code>data</code> object called <code>soybean_cn_sub</code> contained 7,332 genes and three treatment groups, S1, S2, and S3 <span class="citation">(Brown and Hudson 2015)</span>. We can examine the structure of its corresponding <code>dataMetrics</code> object called <code>soybean_cn_sub_metrics</code> as follows:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"soybean_cn_sub_metrics"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_cn_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 3
## $ S1_S2:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma18g00690.1" "Glyma08g22380.1" "Glyma20g30460.1"
##    "Glyma07g09700.1" ...
## ..$ logFC : num [1:7332] 3.14 2.62 2.5 2.37 2.74 ...
## ..$ logCPM: num [1:7332] 8.04 8.05 8.14 8.19 7.93 ...
## ..$ LR : num [1:7332] 29.3 24.8 24 22.7 21 ...
## ..$ PValue: num [1:7332] 6.08e-08 6.50e-07 9.82e-07 1.94e-06 4.57e-06 ...
## ..$ FDR : num [1:7332] 0.000446 0.002384 0.0024 0.003557 0.006697 ...
## $ S1_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g22380.1" "Glyma08g19290.1" "Glyma20g30460.1"
##    "Glyma18g00690.1" ...
## ..$ logFC : num [1:7332] 3.18 3.32 2.44 2.46 3.03 ...
## ..$ logCPM: num [1:7332] 8.05 8.14 8.14 8.04 7.85 ...
## ..$ LR : num [1:7332] 30.4 24.7 23.3 22.5 22.3 ...
## ..$ PValue: num [1:7332] 3.60e-08 6.53e-07 1.42e-06 2.09e-06 2.30e-06 ...
## ..$ FDR : num [1:7332] 0.000264 0.002393 0.003378 0.003378 0.003378 ...
## $ S2_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g14670.3" "Glyma08g14670.2" "Glyma08g19290.1"
##    "Glyma06g46701.1" ...
## ..$ logFC : num [1:7332] 2.66 2.72 2.78 2.19 2.29 ...
## ..$ logCPM: num [1:7332] 7.86 7.82 8.14 7.63 7.84 ...
## ..$ LR : num [1:7332] 16.1 16 14.3 11.8 10.5 ...
## ..$ PValue: num [1:7332] 6.01e-05 6.30e-05 1.57e-04 5.81e-04 1.21e-03 ...
## ..$ FDR : num [1:7332] 0.231 0.231 0.385 1 1 ...</code></pre>
<hr>
</div>
<div id="data-metrics-object-rules" class="section level2">
<h2 class="hasAnchor">
<a href="#data-metrics-object-rules" class="anchor"></a>Data metrics object rules</h2>
<p>As demonstrated in the two examples above, the <code>dataMetrics</code> object must meet the following conditions:</p>
<ul>
<li>Be of type <code>list</code>
</li>
<li>Contain a number of elements equal to the number of pairwise treatment combinations in the <code>data</code> object. For example, the <code>soybean_ir_sub_metrics</code> object contains one list element (“N_P”) and the <code>soybean_cn_sub_metrics</code> object contains three list elements (“S1_S2”, “S1_S3”, “S2_S3”).</li>
<li>Have each list element
<ul>
<li>Be of type <code>data.frame</code>
</li>
<li>Be called in a three-part format (such as “N_P” or “S2_S3”) that matches the Perl expression <code>^[a-zA-Z0-9]+_[a-zA-Z0-9]+</code>, where
<ul>
<li>The first part indicates the first treatment group alphameric name</li>
<li>The second part consists of an underscore "_" to serve as a delimeter</li>
<li>The third part indicates the second treatment group alphameric name</li>
</ul>
</li>
<li>Contain a first column called “ID” of class <code>character</code> consisting of the unique names of the genes</li>
<li>Contain at least one column of class <code>numeric</code> or <code>integer</code> consisting of a quantitative variable. This can be called anything. In the examples above, there are five of such columns called “logFC”, “logCPM”, “LR”, “PValue”, and “FDR”.</li>
</ul>
</li>
</ul>
<p>You can quickly double-check the names of the list elements in your <code>dataMetrics</code> object as follows:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_ir_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "N_P"</code></pre>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_cn_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "S1_S2" "S1_S3" "S2_S3"</code></pre>
<p>If your <code>dataMetrics</code> object does not fit this format, <code>bigPint</code> will likely throw an informative error about why your format was not recognized.</p>
<hr>
</div>
<div id="creating-data-metrics-object" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-data-metrics-object" class="anchor"></a>Creating data metrics object</h2>
<p>If a researcher is using the <code>bigPint</code> package to plot RNA-seq data, then many will create the <code>dataMetrics</code> object using popular RNA-seq packages such as <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span>, and <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> <span class="citation">(Ritchie et al. 2015)</span>. These R packages will output several interesting quantitative variables for each gene in the dataset that can be incorporated into the <code>dataMetrics</code> object. <code>bigPint</code> can then apply thresholds to these variables and create subsets of genes to superimpose. To create numerous <code>bigPint</code> plots with the least effort, we recommend creating a <code>dataMetrics</code> object that contains at least the following column types:</p>
<ul>
<li>Significance level (“PValue”)</li>
<li>Multiple comparison significance level (“FDR”)</li>
<li>Log fold change (“logFC”)</li>
</ul>
<p>Many <code>bigPint</code> plots use “FDR” to determine “significant” genes and subset them as overlay (FDR &lt; 0.05). The <code>bigPint</code> volcano plot uses “PValue” and “logFC”. Naming these columns as above will save you time but the names and the default threshold values can be specified away from default when creating each <code>bigPint</code> plot.</p>
<p>We now provide reproducible code for creating <code>dataMetrics</code> objects with two or three treatment groups using both <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> and <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span>.</p>
<hr>
</div>
<div id="example-two-groups-edger" class="section level2">
<h2 class="hasAnchor">
<a href="#example-two-groups-edger" class="anchor"></a>Example: two groups (edgeR)</h2>
<p>The following example shows how to create the <code>dataMetrics</code> object called <code>soybean_ir_sub_metrics</code>, which was shown in the article <a href="https://lindsayrutter.github.io/bigPint/articles/dataMetrics.html">Data metrics object</a> <span class="citation">(Lauter and Graham 2016)</span>. The dataset from which it is created (<code>soybean_ir_sub</code>) contains only two treatment groups, N and P. In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> package was primarily followed.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">bigPint</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">edgeR</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">data.table</span>)

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">soybean_ir_sub</span>)
<span class="no">data</span> <span class="kw">=</span> <span class="no">soybean_ir_sub</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">data</span>) <span class="kw">=</span> <span class="no">data</span>[,<span class="fl">1</span>]

<span class="no">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span><span class="kw">=</span><span class="no">data</span>[,-<span class="fl">1</span>])
<span class="no">group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>)
<span class="no">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span><span class="kw">=</span><span class="no">y</span>, <span class="kw">group</span><span class="kw">=</span><span class="no">group</span>)
<span class="no">Group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"N"</span>,<span class="fl">3</span>), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"P"</span>,<span class="fl">3</span>)))
<span class="no">design</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~<span class="fl">0</span>+<span class="no">Group</span>, <span class="kw">data</span><span class="kw">=</span><span class="no">y</span>$<span class="no">samples</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">design</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">Group</span>)
<span class="no">y</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span>(<span class="no">y</span>, <span class="no">design</span>)
<span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmFit</a></span>(<span class="no">y</span>, <span class="no">design</span>)

<span class="no">soybean_ir_sub_metrics</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>)-<span class="fl">1</span>)){
  <span class="kw">for</span> (<span class="no">j</span> <span class="kw">in</span> (<span class="no">i</span>+<span class="fl">1</span>):<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>)){
    <span class="no">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>))
    <span class="no">contrast</span>[<span class="no">i</span>]<span class="kw">=</span><span class="fl">1</span>
    <span class="no">contrast</span>[<span class="no">j</span>]<span class="kw">=</span>-<span class="fl">1</span>
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmLRT</a></span>(<span class="no">fit</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="no">contrast</span>)
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span>(<span class="no">lrt</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">y</span><span class="kw">[[</span><span class="fl">1</span>]]))<span class="kw">[[</span><span class="fl">1</span>]]

    <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setDT.html">setDT</a></span>(<span class="no">lrt</span>, <span class="kw">keep.rownames</span> <span class="kw">=</span> <span class="fl">TRUE</span>)[]
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">lrt</span>)[<span class="fl">1</span>] <span class="kw">=</span> <span class="st">"ID"</span>
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">lrt</span>)

    <span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">fit</span>)[<span class="no">i</span>], <span class="st">"_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">fit</span>)[<span class="no">j</span>])]] <span class="kw">&lt;-</span> <span class="no">lrt</span>
  }
}</pre></body></html></div>
<p>We can indeed examine the generated <code>soybean_ir_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_ir_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 1
## $ N_P:'data.frame': 5604 obs. of 6 variables:
## ..$ ID : chr [1:5604] "Glyma.19G168700.Wm82.a2.v1" "Glyma.13G293500.Wm82.a2.v1"
##    "Glyma.05G188700.Wm82.a2.v1" "Glyma.13G173100.Wm82.a2.v1" ...
## ..$ logFC : num [1:5604] -5.92 2.99 -3.51 -3.91 -3.51 ...
## ..$ logCPM: num [1:5604] 7.52 8.08 8.83 8.27 10.19 ...
## ..$ LR : num [1:5604] 266 171 167 157 154 ...
## ..$ PValue: num [1:5604] 9.18e-60 3.65e-39 2.73e-38 6.04e-36 2.58e-35 ...
## ..$ FDR : num [1:5604] 5.14e-56 1.02e-35 5.09e-35 8.46e-33 2.89e-32 ...</code></pre>
<p>And verify that it contains one list element:</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_ir_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "N_P"</code></pre>
<hr>
</div>
<div id="example-three-groups-edger" class="section level2">
<h2 class="hasAnchor">
<a href="#example-three-groups-edger" class="anchor"></a>Example: three groups (edgeR)</h2>
<p>The following example shows how to create the <code>dataMetrics</code> object called <code>soybean_cn_sub_metrics</code>, which was shown in the article <a href="https://lindsayrutter.github.io/bigPint/articles/dataMetrics.html">Data metrics object</a>). The dataset from which it is created (<code>soybean_cn_sub</code>) contains three treatment groups, S1, S2, and S3 <span class="citation">(Brown and Hudson 2015)</span>. In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span> package was primarily followed.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">soybean_cn_sub</span>)
<span class="no">data</span> <span class="kw">=</span> <span class="no">soybean_cn_sub</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">data</span>) <span class="kw">=</span> <span class="no">data</span>[,<span class="fl">1</span>]

<span class="no">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span><span class="kw">=</span><span class="no">data</span>[,-<span class="fl">1</span>])
<span class="no">group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">3</span>)
<span class="no">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span>(<span class="kw">counts</span><span class="kw">=</span><span class="no">y</span>, <span class="kw">group</span><span class="kw">=</span><span class="no">group</span>)
<span class="no">Group</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"S1"</span>,<span class="fl">3</span>), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"S2"</span>,<span class="fl">3</span>), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"S3"</span>,<span class="fl">3</span>)))
<span class="no">design</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~<span class="fl">0</span>+<span class="no">Group</span>, <span class="kw">data</span><span class="kw">=</span><span class="no">y</span>$<span class="no">samples</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">design</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">Group</span>)
<span class="no">y</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span>(<span class="no">y</span>, <span class="no">design</span>)
<span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmFit</a></span>(<span class="no">y</span>, <span class="no">design</span>)

<span class="no">soybean_cn_sub_metrics</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>)-<span class="fl">1</span>)){
  <span class="kw">for</span> (<span class="no">j</span> <span class="kw">in</span> (<span class="no">i</span>+<span class="fl">1</span>):<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>)){
    <span class="no">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">fit</span>))
    <span class="no">contrast</span>[<span class="no">i</span>]<span class="kw">=</span><span class="fl">1</span>
    <span class="no">contrast</span>[<span class="no">j</span>]<span class="kw">=</span>-<span class="fl">1</span>
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmLRT</a></span>(<span class="no">fit</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="no">contrast</span>)
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html">topTags</a></span>(<span class="no">lrt</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">y</span><span class="kw">[[</span><span class="fl">1</span>]]))<span class="kw">[[</span><span class="fl">1</span>]]

    <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setDT.html">setDT</a></span>(<span class="no">lrt</span>, <span class="kw">keep.rownames</span> <span class="kw">=</span> <span class="fl">TRUE</span>)[]
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">lrt</span>)[<span class="fl">1</span>] <span class="kw">=</span> <span class="st">"ID"</span>
    <span class="no">lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">lrt</span>)

    <span class="no">soybean_cn_sub_metrics</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">fit</span>)[<span class="no">i</span>], <span class="st">"_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">fit</span>)[<span class="no">j</span>])]] <span class="kw">&lt;-</span> <span class="no">lrt</span>
  }
}</pre></body></html></div>
<p>We can indeed examine the generated <code>soybean_cn_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_cn_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 3
## $ S1_S2:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma18g00690.1" "Glyma08g22380.1" "Glyma20g30460.1"
##    "Glyma07g09700.1" ...
## ..$ logFC : num [1:7332] 3.14 2.62 2.5 2.37 2.74 ...
## ..$ logCPM: num [1:7332] 8.04 8.05 8.14 8.19 7.93 ...
## ..$ LR : num [1:7332] 29.3 24.8 24 22.7 21 ...
## ..$ PValue: num [1:7332] 6.08e-08 6.50e-07 9.82e-07 1.94e-06 4.57e-06 ...
## ..$ FDR : num [1:7332] 0.000446 0.002384 0.0024 0.003557 0.006697 ...
## $ S1_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g22380.1" "Glyma08g19290.1" "Glyma20g30460.1"
##    "Glyma18g00690.1" ...
## ..$ logFC : num [1:7332] 3.18 3.32 2.44 2.46 3.03 ...
## ..$ logCPM: num [1:7332] 8.05 8.14 8.14 8.04 7.85 ...
## ..$ LR : num [1:7332] 30.4 24.7 23.3 22.5 22.3 ...
## ..$ PValue: num [1:7332] 3.60e-08 6.53e-07 1.42e-06 2.09e-06 2.30e-06 ...
## ..$ FDR : num [1:7332] 0.000264 0.002393 0.003378 0.003378 0.003378 ...
## $ S2_S3:'data.frame': 7332 obs. of 6 variables:
## ..$ ID : chr [1:7332] "Glyma08g14670.3" "Glyma08g14670.2" "Glyma08g19290.1"
##    "Glyma06g46701.1" ...
## ..$ logFC : num [1:7332] 2.66 2.72 2.78 2.19 2.29 ...
## ..$ logCPM: num [1:7332] 7.86 7.82 8.14 7.63 7.84 ...
## ..$ LR : num [1:7332] 16.1 16 14.3 11.8 10.5 ...
## ..$ PValue: num [1:7332] 6.01e-05 6.30e-05 1.57e-04 5.81e-04 1.21e-03 ...
## ..$ FDR : num [1:7332] 0.231 0.231 0.385 1 1 ...</code></pre>
<p>And verify that it contains three list element:</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_cn_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "S1_S2" "S1_S3" "S2_S3"</code></pre>
<hr>
</div>
<div id="example-two-groups-deseq2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-two-groups-deseq2" class="anchor"></a>Example: two groups (DESeq2)</h2>
<p>This example shows how to create a <code>dataMetrics</code> object from (<code>soybean_ir_sub</code>). In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span> package was used.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">DESeq2</span>)

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">soybean_ir_sub</span>)
<span class="no">data</span> <span class="kw">=</span> <span class="no">soybean_ir_sub</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">data</span>) <span class="kw">=</span> <span class="no">data</span>[,<span class="fl">1</span>]
<span class="no">data</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html">as.matrix</a></span>(<span class="no">data</span>[,-<span class="fl">1</span>])

<span class="no">coldata</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">row.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">treatment</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(
  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">function</span> (<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/tstrsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="st">"[.]"</span>))[<span class="fl">1</span>])))
<span class="no">dds</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(<span class="kw">countData</span> <span class="kw">=</span> <span class="no">data</span>, <span class="kw">colData</span> <span class="kw">=</span> <span class="no">coldata</span>,
  <span class="kw">design</span> <span class="kw">=</span> ~ <span class="no">treatment</span>)
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds</span>)

<span class="no">uTreat</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">function</span> (<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/tstrsplit.html">strsplit</a></span>(
  <span class="no">x</span>, <span class="st">"[.]"</span>))[<span class="fl">1</span>])))
<span class="no">soybean_ir_sub_metrics</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">uTreat</span>)-<span class="fl">1</span>)){
    <span class="kw">for</span> (<span class="no">j</span> <span class="kw">in</span> (<span class="no">i</span>+<span class="fl">1</span>):<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">uTreat</span>)){
        <span class="no">res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"treatment"</span>, <span class="no">uTreat</span>[<span class="no">i</span>], <span class="no">uTreat</span>[<span class="no">j</span>]))
        <span class="no">metrics</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">res</span>@<span class="kw">listData</span>)
        <span class="no">metrics</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span>(<span class="kw">ID</span> <span class="kw">=</span> <span class="no">res</span>@<span class="kw">rownames</span>, <span class="no">metrics</span>)
        <span class="no">metrics</span>$<span class="no">ID</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">metrics</span>$<span class="no">ID</span>)
        <span class="no">metrics</span> <span class="kw">&lt;-</span> <span class="no">metrics</span>[<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span>(<span class="no">metrics</span>$<span class="no">padj</span>), ]
        <span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">uTreat</span>[<span class="no">i</span>], <span class="st">"_"</span>, <span class="no">uTreat</span>[<span class="no">j</span>])]] <span class="kw">&lt;-</span> <span class="no">metrics</span>
    }
}</pre></body></html></div>
<p>By default, <code>DESeq2</code> gives output with variables called “pvalue”, “padj”, and “log2FoldChange”. Various functions in <code>bigPint</code> expect column names like “FDR”, “logFC”, and “PValue” respectively in the <code>dataMetrics</code> object. That can be modified manually using the <code>threshVar</code> input parameter each time creating a plot. But it is easier to simply rename these parameters from the start in the <code>dataMetrics</code> object.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="kw">for</span> (<span class="no">df</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">soybean_ir_sub_metrics</span>))){
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"pvalue"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"PValue"</span>
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"padj"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"FDR"</span>
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"log2FoldChange"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"logFC"</span>
}</pre></body></html></div>
<p>We can indeed examine the generated <code>soybean_ir_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_ir_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 1
## $ N_P:'data.frame': 5604 obs. of 7 variables:
## ..$ ID : chr [1:5604] "Glyma.11G141800.Wm82.a2.v1" "Glyma.11G044800.Wm82.a2.v1"
##    "Glyma.05G188700.Wm82.a2.v1" "Glyma.13G173100.Wm82.a2.v1" ...
## ..$ baseMean: num [1:5604] 778 795 302 204 420 ...
## ..$ logFC : num [1:5604] -3.54 2.75 -3.53 -3.94 -3.1 ...
## ..$ lfcSE : num [1:5604] 0.249 0.207 0.267 0.31 0.244 ...
## ..$ stat : num [1:5604] -14.2 13.3 -13.2 -12.7 -12.7 ...
## ..$ PValue : num [1:5604] 5.49e-46 3.49e-40 5.34e-40 4.80e-37 5.76e-37 ...
## ..$ FDR : num [1:5604] 1.95e-42 6.20e-37 6.32e-37 4.09e-34 4.09e-34 ...</code></pre>
<p>And verify that it contains one list element:</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_ir_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "N_P"</code></pre>
<hr>
</div>
<div id="example-three-groups-deseq2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-three-groups-deseq2" class="anchor"></a>Example: three groups (DESeq2)</h2>
<p>This example shows how to create a <code>dataMetrics</code> object from (<code>soybean_cn_sub</code>). In this case, the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span> package was used. The <code>DESeq</code> package expects a count table that contains integers and has gene-wise dispersion estimates larger than two orders of magnitude from the minimum value. To fit this requirement just for this didactic exercise, we multiply each value by ten and perform a <code><a href="https://rdrr.io/r/base/Round.html">ceiling()</a></code> function.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">soybean_cn_sub</span>)
<span class="no">data</span> <span class="kw">=</span> <span class="no">soybean_cn_sub</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">data</span>) <span class="kw">=</span> <span class="no">data</span>[,<span class="fl">1</span>]
<span class="no">data</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Round.html">ceiling</a></span>(<span class="no">data</span>[,-<span class="fl">1</span>] * <span class="fl">10</span>))

<span class="no">coldata</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">row.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">treatment</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(
  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">function</span> (<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/tstrsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="st">"[.]"</span>))[<span class="fl">1</span>])))
<span class="no">dds</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span>(<span class="kw">countData</span> <span class="kw">=</span> <span class="no">data</span>, <span class="kw">colData</span> <span class="kw">=</span> <span class="no">coldata</span>,
  <span class="kw">design</span> <span class="kw">=</span> ~ <span class="no">treatment</span>)
<span class="no">dds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dds</span>)

<span class="no">uTreat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">data</span>), <span class="kw">function</span> (<span class="no">x</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/tstrsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="st">"[.]"</span>))[<span class="fl">1</span>])))
<span class="no">soybean_cn_sub_metrics</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">uTreat</span>)-<span class="fl">1</span>)){
    <span class="kw">for</span> (<span class="no">j</span> <span class="kw">in</span> (<span class="no">i</span>+<span class="fl">1</span>):<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">uTreat</span>)){
        <span class="no">res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dds</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"treatment"</span>, <span class="no">uTreat</span>[<span class="no">i</span>], <span class="no">uTreat</span>[<span class="no">j</span>]))
        <span class="no">metrics</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">res</span>@<span class="kw">listData</span>)
        <span class="no">metrics</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span>(<span class="kw">ID</span> <span class="kw">=</span> <span class="no">res</span>@<span class="kw">rownames</span>, <span class="no">metrics</span>)
        <span class="no">metrics</span>$<span class="no">ID</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">metrics</span>$<span class="no">ID</span>)
        <span class="no">metrics</span> <span class="kw">&lt;-</span> <span class="no">metrics</span>[<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span>(<span class="no">metrics</span>$<span class="no">padj</span>), ]
        <span class="no">soybean_cn_sub_metrics</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">uTreat</span>[<span class="no">i</span>], <span class="st">"_"</span>, <span class="no">uTreat</span>[<span class="no">j</span>])]] <span class="kw">&lt;-</span> <span class="no">metrics</span>
    }
}</pre></body></html></div>
<p>By default, <code>DESeq2</code> gives output with variables called “pvalue”, “padj”, and “log2FoldChange”. Various functions in <code>bigPint</code> expect column names like “FDR”, “logFC”, and “PValue” respectively in the <code>dataMetrics</code> object. That can be modified manually using the <code>threshVar</code> input parameter each time creating a plot. But it is easier to simply rename these parameters from the start in the <code>dataMetrics</code> object.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="kw">for</span> (<span class="no">df</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">soybean_ir_sub_metrics</span>))){
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"pvalue"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"PValue"</span>
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"padj"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"FDR"</span>
    <span class="no">whichPadj</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])<span class="kw">==</span><span class="st">"log2FoldChange"</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">soybean_ir_sub_metrics</span><span class="kw">[[</span><span class="no">df</span>]])[<span class="no">whichPadj</span>] <span class="kw">=</span> <span class="st">"logFC"</span>
}</pre></body></html></div>
<p>We can indeed examine the generated <code>soybean_cn_sub_metrics</code> object as follows:</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">soybean_cn_sub_metrics</span>, <span class="kw">strict.width</span> <span class="kw">=</span> <span class="st">"wrap"</span>)</pre></body></html></div>
<pre><code>## List of 3
## $ S1_S2:'data.frame': 7332 obs. of 7 variables:
## ..$ ID : chr [1:7332] "Glyma18g00690.1" "Glyma03g29150.1" "Glyma05g27450.2"
##    "Glyma10g31780.1" ...
## ..$ baseMean : num [1:7332] 50.4 33 46.3 45.3 29.8 ...
## ..$ log2FoldChange: num [1:7332] 3.22 3.14 -2.89 2.83 2.99 ...
## ..$ lfcSE : num [1:7332] 0.427 0.419 0.4 0.395 0.42 ...
## ..$ stat : num [1:7332] 7.55 7.5 -7.22 7.16 7.12 ...
## ..$ pvalue : num [1:7332] 4.30e-14 6.35e-14 5.03e-13 7.92e-13 1.06e-12 ...
## ..$ padj : num [1:7332] 1.92e-10 1.92e-10 1.01e-09 1.20e-09 1.28e-09 ...
## $ S1_S3:'data.frame': 7332 obs. of 7 variables:
## ..$ ID : chr [1:7332] "Glyma08g19290.1" "Glyma08g22380.1" "Glyma16g08810.1"
##    "Glyma04g37510.1" ...
## ..$ baseMean : num [1:7332] 55.4 50.8 34 39.1 45.7 ...
## ..$ log2FoldChange: num [1:7332] 3.39 3.26 3.17 -2.99 -2.9 ...
## ..$ lfcSE : num [1:7332] 0.391 0.404 0.424 0.405 0.4 ...
## ..$ stat : num [1:7332] 8.67 8.06 7.49 -7.4 -7.26 ...
## ..$ pvalue : num [1:7332] 4.29e-18 7.39e-16 7.15e-14 1.40e-13 3.90e-13 ...
## ..$ padj : num [1:7332] 2.66e-14 2.29e-12 1.48e-10 2.17e-10 4.83e-10 ...
## $ S2_S3:'data.frame': 7332 obs. of 7 variables:
## ..$ ID : chr [1:7332] "Glyma08g19290.1" "Glyma08g14670.2" "Glyma08g14670.3"
##    "Glyma02g06650.1" ...
## ..$ baseMean : num [1:7332] 55.4 40.7 42.3 41.4 45.7 ...
## ..$ log2FoldChange: num [1:7332] 2.81 2.77 2.68 2.31 1.92 ...
## ..$ lfcSE : num [1:7332] 0.396 0.395 0.387 0.4 0.342 ...
## ..$ stat : num [1:7332] 7.1 7.02 6.94 5.77 5.62 ...
## ..$ pvalue : num [1:7332] 1.28e-12 2.28e-12 3.88e-12 8.01e-09 1.94e-08 ...
## ..$ padj : num [1:7332] 6.91e-09 6.91e-09 7.83e-09 1.21e-05 2.35e-05 ...</code></pre>
<p>And verify that it contains three list element:</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">soybean_cn_sub_metrics</span>)</pre></body></html></div>
<pre><code>## [1] "S1_S2" "S1_S3" "S2_S3"</code></pre>
<hr>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-brown2015developmental">
<p>Brown, Anne V., and Karen A. Hudson. 2015. “Developmental Profiling of Gene Expression in Soybean Trifoliate Leaves and Cotyledons.” <em>BMC Plant Biology</em> 15 (1): 169.</p>
</div>
<div id="ref-soybeanIR">
<p>Lauter, AN Moran, and MA Graham. 2016. “NCBI Sra Bioproject Accession: PRJNA318409.”</p>
</div>
<div id="ref-love2014moderated">
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12): 550.</p>
</div>
<div id="ref-ritchie2015limma">
<p>Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. “Limma Powers Differential Expression Analyses for Rna-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47–e47.</p>
</div>
<div id="ref-robinson2010edger">
<p>Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. “EdgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.” <em>Bioinformatics</em> 26 (1): 139–40.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/lindsayrutter">Lindsay Rutter</a>, <a href="https://github.com/dicook">Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
