<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data object • bigPint</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Data object">
<meta property="og:description" content="">
<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/summarizedExperiment.html">SummarizedExperiment object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    manuscripts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/honeybee.html">Research article</a>
    </li>
    <li>
      <a href="../articles/methods.html">Methodology article</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Data object</h1>
                        <h4 class="author">Lindsay Rutter</h4>
                        
            <h4 class="date">2020-02-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lindsayrutter/bigPint/blob/master/vignettes/data.Rmd"><code>vignettes/data.Rmd</code></a></small>
      <div class="hidden name"><code>data.Rmd</code></div>

    </div>

    
    
<div id="about-data-object" class="section level2">
<h2 class="hasAnchor">
<a href="#about-data-object" class="anchor"></a>About data object</h2>
<p>All functions in the <code>bigPint</code> package require an input parameter called <code>data</code>, which should be a data frame that contains the full dataset of interest. If a researcher is using the package to visualize RNA-seq data, then this <code>data</code> object should be a count table that contains the read counts for all genes of interest.</p>
<hr>
</div>
<div id="example-two-treatments" class="section level2">
<h2 class="hasAnchor">
<a href="#example-two-treatments" class="anchor"></a>Example: two treatments</h2>
<p>The <code>data</code> object requires the same particular data frame format for all <code>bigPint</code> functions. There should be <span class="math inline">\(n\)</span> rows in the data frame, where <span class="math inline">\(n\)</span> is the number of genes. There should be <span class="math inline">\(p + 1\)</span> columns in the data frame, where <span class="math inline">\(p\)</span> is the number of samples. The first column contains the genes names and the rest of the columns should contain the read counts for all samples of interest. An example of this format is shown below:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(bigPint)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"soybean_ir_sub"</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(soybean_ir_sub)</span></code></pre></div>
<pre><code>##                               ID N.1 N.2 N.3 P.1 P.2 P.3
## 14881 Glyma.06G158700.Wm82.a2.v1  48  28  15  29  16   8
## 20855 Glyma.08G156000.Wm82.a2.v1   0   0   0   0   0   0
## 32104 Glyma.12G070600.Wm82.a2.v1   8  11  11   5   8   7
## 50897 Glyma.19G045200.Wm82.a2.v1 200 192 187 186 193 183
## 11303 Glyma.05G050000.Wm82.a2.v1   0   0   0   0   0   0
## 50345 Glyma.18G292300.Wm82.a2.v1 583 669 497 419 467 426</code></pre>
<p>We can also examine the structure of an example <code>data</code> object as follows:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(soybean_ir_sub, <span class="dt">strict.width =</span> <span class="st">"wrap"</span>)</span></code></pre></div>
<pre><code>## 'data.frame':    5604 obs. of  7 variables:
## $ ID : chr "Glyma.06G158700.Wm82.a2.v1" "Glyma.08G156000.Wm82.a2.v1"
##    "Glyma.12G070600.Wm82.a2.v1" "Glyma.19G045200.Wm82.a2.v1" ...
## $ N.1: int 48 0 8 200 0 583 22 52 1 73 ...
## $ N.2: int 28 0 11 192 0 669 34 42 0 120 ...
## $ N.3: int 15 0 11 187 0 497 19 44 3 59 ...
## $ P.1: int 29 0 5 186 0 419 11 46 0 98 ...
## $ P.2: int 16 0 8 193 0 467 21 54 0 106 ...
## $ P.3: int 8 0 7 183 0 426 11 42 2 86 ...</code></pre>
<p>This example dataset contains 5,604 genes and six samples <span class="citation">(Lauter and Graham 2016)</span>. There are two treatment groups, N and P. Each treatment group contains three replicates.</p>
<hr>
</div>
<div id="data-object-rules" class="section level2">
<h2 class="hasAnchor">
<a href="#data-object-rules" class="anchor"></a>Data object rules</h2>
<p>As demonstrated above, the <code>data</code> object must meet the following conditions:</p>
<ul>
<li>Be of type <code>data.frame</code>
</li>
<li>Contain at least two treatment groups and at least two replicates per treatment group</li>
<li>Its first column must
<ul>
<li>Be called “ID”</li>
<li>Be of class <code>character</code>
</li>
<li>Contain the names of the genes (or a unique set of names in general)</li>
</ul>
</li>
<li>Each of the rest of its columns must
<ul>
<li>Contain the read counts for a given sample (or quantitative values in general)</li>
<li>Be of class <code>integer</code> or <code>numeric</code>
</li>
<li>Be called in a three-part format (such as “A.3” or “S4.1”) that matches the Perl expression <code>^[a-zA-Z0-9]+\\.[0-9]+</code>, where
<ul>
<li>The first part indicates the treatment group name and must contain alphanumeric characters. Examples include “A”, “AR”, and “A9”</li>
<li>The second part consists of a dot “.” to serve as a delimeter</li>
<li>The third part indicates the replicate number and must consist of numbers</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>It is important that the names of all columns except the first follow the three-part format delineated above. All functions in the <code>bigPint</code> package require this format to successfully produce plots. If your <code>data</code> object does not fit this format, <code>bigPint</code> will likely throw an informative error about why your format was not recognized.</p>
<hr>
</div>
<div id="example-three-treatments" class="section level2">
<h2 class="hasAnchor">
<a href="#example-three-treatments" class="anchor"></a>Example: three treatments</h2>
<p>Note that the <code>data</code> object can contain more than two treatment groups. In this case, the <code>bigPint</code> software will automatically create plots for all pairs of treatment groups. An example of this type of dataset is provided in the <code>bigPint</code> package and can accessed as follows:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(soybean_cn_sub)</span></code></pre></div>
<p>This example dataset contains 7,332 genes and nine samples <span class="citation">(Brown and Hudson 2015)</span>. There are three treatment groups, S1, S2, and S3. Each treatment group contains three replicates. In such cases where the <code>data</code> object contains more than two treatment groups, all functions in the <code>bigPint</code> package (except <code><a href="../reference/plotSMApp.html">plotSMApp()</a></code>) will automatically produce a plot for each pairwise combination of treatment groups.</p>
<p>For example, <code>bigPint</code> functions will produce plots for S1 versus S2, S1 versus S3, and S2 versus S3 in this case. The same could be accomplished (although less efficiently) by separating the dataset into three separate datasets and running a <code>bigPint</code> function of interest on each of them individually.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(dplyr)</span>
<span id="cb6-2"><a href="#cb6-2"></a>soybean_cn_sub_S1S2 &lt;-<span class="st"> </span>soybean_cn_sub <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="st">"ID"</span>, <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S1"</span>), <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S2"</span>))</span>
<span id="cb6-3"><a href="#cb6-3"></a>soybean_cn_sub_S1S3 &lt;-<span class="st"> </span>soybean_cn_sub <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="st">"ID"</span>, <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S1"</span>), <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S3"</span>))</span>
<span id="cb6-4"><a href="#cb6-4"></a>soybean_cn_sub_S2S3 &lt;-<span class="st"> </span>soybean_cn_sub <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="st">"ID"</span>, <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S2"</span>), <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">contains</a></span>(<span class="st">"S3"</span>))</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(soybean_cn_sub_S1S2, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                     ID      S1.1     S1.2     S1.3     S2.1     S2.2     S2.3
## 19468  Glyma06g12670.1 0.8024444 2.708884 1.763407 7.716099 6.581990 7.003538
## 27284  Glyma08g12390.2 4.7687202 5.235777 5.166631 3.823472 3.566863 3.295619
## 42001 Glyma12g02076.11 3.1899340 2.902131 2.906502 3.100206 3.284326 3.295619</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(soybean_cn_sub_S1S3, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                     ID      S1.1     S1.2     S1.3     S3.1     S3.2     S3.3
## 19468  Glyma06g12670.1 0.8024444 2.708884 1.763407 8.556732 8.367593 8.389347
## 27284  Glyma08g12390.2 4.7687202 5.235777 5.166631 3.669489 4.031427 4.269312
## 42001 Glyma12g02076.11 3.1899340 2.902131 2.906502 3.364437 2.731105 3.255649</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(soybean_cn_sub_S2S3, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                     ID     S2.1     S2.2     S2.3     S3.1     S3.2     S3.3
## 19468  Glyma06g12670.1 7.716099 6.581990 7.003538 8.556732 8.367593 8.389347
## 27284  Glyma08g12390.2 3.823472 3.566863 3.295619 3.669489 4.031427 4.269312
## 42001 Glyma12g02076.11 3.100206 3.284326 3.295619 3.364437 2.731105 3.255649</code></pre>
<hr>
</div>
<div id="preprocessing-of-data-object" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing-of-data-object" class="anchor"></a>Preprocessing of data object</h2>
<p>Some popular RNA-seq analysis packages (such as <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span>, and <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> <span class="citation">(Ritchie et al. 2015)</span>) advise researchers to perform certain preprocessing steps to their data, such as filtering the genes, normalizing their read counts, and standardizing their read counts before visualization. Researchers can use datasets whether or not they have been filtered, normalized, and standardized for setting the <code>data</code> object in the <code>bigPint</code> package. If they wish, they can use <code>bigPint</code> plots to investigate how their dataset changes after filters, normalizations, and standardizations.</p>
<hr>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-brown2015developmental">
<p>Brown, Anne V., and Karen A. Hudson. 2015. “Developmental Profiling of Gene Expression in Soybean Trifoliate Leaves and Cotyledons.” <em>BMC Plant Biology</em> 15 (1): 169.</p>
</div>
<div id="ref-soybeanIR">
<p>Lauter, AN Moran, and MA Graham. 2016. “NCBI Sra Bioproject Accession: PRJNA318409.”</p>
</div>
<div id="ref-love2014moderated">
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12): 550.</p>
</div>
<div id="ref-ritchie2015limma">
<p>Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. “Limma Powers Differential Expression Analyses for Rna-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47–e47.</p>
</div>
<div id="ref-robinson2010edger">
<p>Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. “EdgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.” <em>Bioinformatics</em> 26 (1): 139–40.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#about-data-object">About data object</a></li>
      <li><a href="#example-two-treatments">Example: two treatments</a></li>
      <li><a href="#data-object-rules">Data object rules</a></li>
      <li><a href="#example-three-treatments">Example: three treatments</a></li>
      <li><a href="#preprocessing-of-data-object">Preprocessing of data object</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/lindsayrutter">Lindsay Rutter</a>, <a href="https://github.com/dicook">Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
