<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to bigPint plots • bigPint</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Introduction to bigPint plots">
<meta property="og:description" content="">
<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/createDataMetrics.html">Creating data metrics object</a>
    </li>
    <li>
      <a href="../articles/superimposeData.html">Alternatives to data metrics object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    manuscripts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/honeybee.html">Research article</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to bigPint plots</h1>
                        <h4 class="author">Lindsay Rutter</h4>
                        
            <h4 class="date">2020-02-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lindsayrutter/bigPint/blob/master/vignettes/plotIntro.Rmd"><code>vignettes/plotIntro.Rmd</code></a></small>
      <div class="hidden name"><code>plotIntro.Rmd</code></div>

    </div>

    
    
<div id="interpreting-plots-in-this-package" class="section level2">
<h2 class="hasAnchor">
<a href="#interpreting-plots-in-this-package" class="anchor"></a>Interpreting plots in this package</h2>
<p>The <code>bigPint</code> package focuses on four plotting techniques that can uncover a wealth of information in biased multivariate data: scatterplot matrices, litre plots, parallel coordinate plots, and volcano plots. These tools can detect normalization problems, significance designation problems, and common errors in the analysis pipelines. Below, we briefly describe the interpretation of the four visualization methods. While these plots can be used for any multivariate dataset, we focus on the interpretation for RNA-sequencing data. For this purpose, we use a clean dataset <a href="https://lindsayrutter.github.io/bigPint/reference/soybean_cn_sub.html"><code>soybean_cn_sub</code></a> derived from soybean cotyledon at different time points. We will consider two treatment groups (S1 and S2) that each have three replicates <span class="citation">(Brown and Hudson 2015)</span>.</p>
<hr>
</div>
<div id="scatterplot-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#scatterplot-matrices" class="anchor"></a>Scatterplot matrices</h2>
<p>A scatterplot matrix is a useful multivariate plotting tool that draws read counts across all genes and samples. It represents each case (gene) as a point in each scatterplot. Users can quickly evaluate the association between multiple variables in a manner that is unique from most common practices.</p>
<p>Within each scatterplot, most genes should fall along the <em>x=y</em> line (in red) as only a small proportion of them should show differential expression. However, the nine treatment scatterplots (bottom-left corner of the matrix) should have more spread around the <em>x=y</em> relationship than the six replicate scatterplots (top-left and bottom-right corners of the matrix); this is because some genes will show larger variability between treatments than between replicates.</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-1-1.png" width="700.0000032"></p>
<p>Users can easily overlay the differentially expressed genes (DEGs) returned from a given model onto the scatterplot matrix. This allows users to immediately assess the associations between samples that ``significant genes’’ show compared to that shown by <em>all</em> genes in the dataset. The figure below shows DEGs superimposed as pink points. We expect for DEGs to fall along the <em>x=y</em> line for scatterplots between replicates and deviate from the <em>x=y</em> line for scatterplots between treatment groups, which the figure below confirms. If a user does not observe this expected pattern of DEGs, then the DEG calls from the model may need to be scrutinized further.</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-2-1.png" width="700.0000032"></p>
<p>We note that the DEG superimposition on the above scatterplot matrix may be hard to interpret. This may be because of the few number of DEGs. In cases like this, it can be useful to standardize the data before plotting. Below, we now standardize the data so that each gene has a mean read count of 0 and standard deviation of 1 across its samples. We can verify more clearly now that the DEGs appear as expected: They fall along the <em>x=y</em> line in replicate scatterplots but deviate from the <em>x=y</em> line in treatment scatterplots.</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-3-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="litre-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#litre-plots" class="anchor"></a>Litre plots</h2>
<p>Above, we showed how to view DEGs onto the Cartesian coordinates of the scatterplot matrix. Unfortunately, the scatterplot matrix becomes limited in cases with a large number of samples because we then must cram too many small scatterplots into the matrix. Researchers could benefit from additional visualization tools that allow them to verify individual DEGs returned from a model. In light of this, we developed a plot that allows users to visualize <em>one</em> DEG of interest onto the Cartesian coordinates of <em>one</em> scatterplot matrix. The plot we introduce here is an extension of the “replicate line plot” that was developed by researchers using microarray data with two replicates<span class="citation">(Cook et al. 2007)</span>. Our new plot can be applied to datasets with two or more replicates. We call this new plot a <strong>repLIcate TREatment (“litre”) plot</strong>.</p>
<p>In the litre plot, each gene is plotted for each combination of replicates between treatment groups. For example, there are nine ways to pair a replicate from one treatment group with a replicate from the other treatment group in the soybean cotyledon dataset (S1.1 and S2.1, S1.1 and S2.2, S1.1 and S2.3, S1.2 and S2.1, S1.2. and S2.2, S1.2 and S2.3, S1.3 and S2.1, S1.3 and S2.2, and S1.3 and S2.3). Therefore, each gene in this data would be plotted as nine points in the litre plot. With 7,332 genes in this data, we would need to plot 65,988 points. This can slow down the drawing process and create overplotting issues (where it is difficult to determine the number of points in a given area because so many points have been plotted over each other in that specific area). As a result, we use hexagon bins to summarize this massive information. Once the background of hexagons has been drawn to give us an idea of the distribution of all between-treatment sample pair combinations for all genes, the user can superimpose the nine points of one gene of interest.</p>
<p>Below is an example litre plot. Notice from the legend that a large number of the replicate treatment values fall along the <em>x=y</em> line because we expect most genes in the data to not show differential expression. The specific gene we are superimposing is a DEG. In fact, it is the gene with the lowest FDR value returned from a model. The nine overlaid yellow points are superimposed in a manner we would expect from a DEG: They are located far from the <em>x=y</em> line (difference between treatments) and are close to each other (similarity between replicates).</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-4-1.png" width="700.0000032"></p>
<p>In contrast, below is an example litre plot for a gene that had a large FDR value in this dataset and was far from being called a DEG in the model. The nine overlaid yellow points are superimposed in a manner we would expect from a non-significant gene: They are far from each other (dissimilarity between replicates) despite having low counts and actually overlap the <em>x=y</em> line (showing no difference between treatments).</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-5-1.png" width="700.0000032"></p>
<p>Note that, again, we can standardize the data before creating litre plots. This can sometimes magnify subtle patterns for easier interpretation. Here, we standardize so that each gene has a mean read count of 0 and standard deviation of 1 across its samples. The resulting <em>standardized</em> DEG litre plot is shown below:</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-6-1.png" width="700.0000032"></p>
<p>And the resulting <em>standardized</em> litre plot for the non-DEG is shown below:</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-7-1.png" width="700.0000032"></p>
<hr>
</div>
<div id="parallel-coordinate-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#parallel-coordinate-plots" class="anchor"></a>Parallel coordinate plots</h2>
<p>Parallel coordinate plots are crucial to visually verify the relationships between samples in multivariate data. A parallel coordinate plot draws each case (gene) as a line. Connections between samples with positive correlations are flat, and connections between samples with negative correlations are crossed. As an example, we show below the DEGs from the soybean cotyledon dataset superimposed as orange lines on a side-by-side boxplot. The side-by-side boxplot shows the read count distribution across <em>all</em> genes in the dataset. Together, the user can quickly ascertain how the read counts of the DEGs (orange parallel coordinate lines) compare to the dataset as a whole (side-by-side boxplot). Note that we optionally chose to standardize the dataset before plotting so that each gene has a mean of zero and a standard deviation of one across its samples <span class="citation">(Chandrasekhar, Thangavel, and Elayaraja 2012)</span>.</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-8-1.png" width="700.0000032"></p>
<p>We note that in the plot above, the 16 DEGs appear as expected: They have consistent (level) replicates but cross between the treatment groups indicating differences in response to treatments. For didactic purposes, we next repeat this same procedure, only now we examine the subset of 16 genes from this dataset that had the highest FDR values from the model (i.e. that demonstrated the least similarity to DEGs).</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-9-1.png" width="700.0000032"></p>
<p>The parallel coordinate plot here confirms what we expect for genes with high FDR values: There is inconsistency between replicates and no observable difference between treatment groups. The article <a href="https://lindsayrutter.github.io/bigPint/articles/clusters.html">Hierarchical clustering</a> demonstrates the use of the parallel coordinate plot to understand general patterns in the dataset.</p>
<hr>
</div>
<div id="volcano-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h2>
<p>The volcano plot is another useful form of the scatterplot that has become increasingly common for the analysis of omic datasets <span class="citation">(Ritchie et al. 2015)</span>. One axis maps a measure of statistical significance from a statistical test (such as p-value) and the other axis maps a magnitude of change. As a result, the user can quickly examine how statistical significance and magnitude change associate in the dataset. Importantly, they can also determine which data points (genes) demonstrate both statistical significance and large magnitude change.</p>
<p>Below, we superimpose the 16 DEGs from the soybean cotyledon dataset as red points onto a volcano plot. We again use hexagon bins to summarize the full dataset and reduce overplotting problems. We can confirm that these 16 DEGs in general have both high magnitude and statistical significance.</p>
<p><img src="plotIntro_files/figure-html/unnamed-chunk-10-1.png" width="700.0000032"></p>
<p>While volcano plots are common in other RNA-seq visualization packages, we also include an interactive version in the <code>bigPint</code> package, which can be viewed in the article <a href="https://lindsayrutter.github.io/bigPint/articles/interactive.html">Producing interactive plots</a> or on the help manual for the interactive volcano function, <a href="https://lindsayrutter.github.io/bigPint/reference/plotVolcanoApp.html">plotVolcanoApp()</a>.</p>
<hr>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-brown2015developmental">
<p>Brown, Anne V., and Karen A. Hudson. 2015. “Developmental Profiling of Gene Expression in Soybean Trifoliate Leaves and Cotyledons.” <em>BMC Plant Biology</em> 15 (1): 169.</p>
</div>
<div id="ref-Chandrasekhar">
<p>Chandrasekhar, T, K Thangavel, and E Elayaraja. 2012. “Effective Clustering Algorithms for Gene Expression Data.” <em>International Journal of Computer Applications</em> 1201: 4914.</p>
</div>
<div id="ref-litre">
<p>Cook, D, H Hofmann, EK Lee, H Yang, B Nikolau, and E Wurtele. 2007. “Exploring Gene Expression Data, Using Plots.” <em>Journal of Data Science</em> 5: 151–82.</p>
</div>
<div id="ref-ritchie2015limma">
<p>Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. “Limma Powers Differential Expression Analyses for Rna-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47–e47.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#interpreting-plots-in-this-package">Interpreting plots in this package</a></li>
      <li><a href="#scatterplot-matrices">Scatterplot matrices</a></li>
      <li><a href="#litre-plots">Litre plots</a></li>
      <li><a href="#parallel-coordinate-plots">Parallel coordinate plots</a></li>
      <li><a href="#volcano-plots">Volcano plots</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/lindsayrutter">Lindsay Rutter</a>, <a href="https://github.com/dicook">Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
