<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot static parallel coordinate clusters — plotClusters • bigPint</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Plot static parallel coordinate clusters — plotClusters" />

<meta property="og:description" content="Perform hierarchical clustering analysis and visualize 
results with parallel coordinate plots. Optionally, save gene IDs within 
each cluster to .rds files for later use." />

<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/createDataMetrics.html">Creating data metrics object</a>
    </li>
    <li>
      <a href="../articles/superimposeData.html">Alternatives to data metrics object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot static parallel coordinate clusters</h1>
    <small class="dont-index">Source: <a href='https://github.com/lindsayrutter/bigPint/blob/master/R/plotClusters.R'><code>R/plotClusters.R</code></a></small>
    <div class="hidden name"><code>plotClusters.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Perform hierarchical clustering analysis and visualize 
results with parallel coordinate plots. Optionally, save gene IDs within 
each cluster to .rds files for later use.</p>
    
    </div>

    <pre class="usage"><span class='fu'>plotClusters</span>(<span class='no'>data</span>, <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>geneList</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>geneLists</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>threshVar</span> <span class='kw'>=</span> <span class='st'>"FDR"</span>, <span class='kw'>threshVal</span> <span class='kw'>=</span> <span class='fl'>0.05</span>,
  <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>nC</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>colList</span> <span class='kw'>=</span> <span class='fu'>rainbow</span>(<span class='no'>nC</span>),
  <span class='kw'>aggMethod</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"ward.D"</span>, <span class='st'>"ward.D2"</span>, <span class='st'>"single"</span>, <span class='st'>"complete"</span>, <span class='st'>"average"</span>,
  <span class='st'>"mcquitty"</span>, <span class='st'>"median"</span>, <span class='st'>"centroid"</span>), <span class='kw'>yAxisLabel</span> <span class='kw'>=</span> <span class='st'>"Count"</span>,
  <span class='kw'>xAxisLabel</span> <span class='kw'>=</span> <span class='st'>"Sample"</span>, <span class='kw'>lineSize</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>lineAlpha</span> <span class='kw'>=</span> <span class='fl'>0.5</span>,
  <span class='kw'>vxAxis</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>outDir</span> <span class='kw'>=</span> <span class='fu'>tempdir</span>(), <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>DATA FRAME | Read counts</p></td>
    </tr>
    <tr>
      <th>dataMetrics</th>
      <td><p>LIST | Differential expression metrics; default NULL</p></td>
    </tr>
    <tr>
      <th>geneList</th>
      <td><p>CHARACTER ARRAY | Array of ID values of genes to be drawn 
from data as parallel coordinate lines. Use this parameter if you have 
predetermined genes to be drawn. These genes will be clustered. Otherwise, 
use dataMetrics, threshVar, and threshVal to create clusters to be
overlaid as parallel coordinate lines; default NULL. See package website
for examples</p></td>
    </tr>
    <tr>
      <th>geneLists</th>
      <td><p>LIST | List of ID values of genes already clustered to be #' drawn from data as parallel coordinate lines. Each list item is an array
of genes ID values that are already grouped as a cluster. Unlike the 
singular geneList object, the plural geneLists object is not be clustered.
If you instead wish to cluster genes, use dataMetrics, threshVar, and 
threshVal or geneList to create clusters to be overlaid as parallel
coordinate lines; default NULL. See package website for examples</p></td>
    </tr>
    <tr>
      <th>threshVar</th>
      <td><p>CHARACTER STRING | Name of column in dataMetrics object 
that is used to threshold significance; default "FDR"</p></td>
    </tr>
    <tr>
      <th>threshVal</th>
      <td><p>INTEGER | Maximum value to threshold significance from 
threshVar object; default 0.05</p></td>
    </tr>
    <tr>
      <th>clusterAllData</th>
      <td><p>BOOLEAN [TRUE | FALSE] | Create clusters based on 
the whole dataset and then assign significant genes to those clusters; 
default is TRUE. If FALSE, create clusters based on just the significant 
genes. With either option, the side-by-side boxplot will represent the 
whole dataset (from data input) and the parallel coordinate lines will 
represent only the significant genes (those that pass threshVal for 
threshVar)</p></td>
    </tr>
    <tr>
      <th>nC</th>
      <td><p>INTEGER | Number of clusters; default 4</p></td>
    </tr>
    <tr>
      <th>colList</th>
      <td><p>CHARACTER ARRAY | List of colors for each cluster; default 
is rainbow(nC)</p></td>
    </tr>
    <tr>
      <th>aggMethod</th>
      <td><p>CHARACTER STRING ["ward.D" | "ward.D2" | "single" | 
"complete" | "average" | "mcquitty" | "median" | "centroid"] | The 
agglomeration method to be used in the hierarchical clustering; default 
"ward.D"</p></td>
    </tr>
    <tr>
      <th>yAxisLabel</th>
      <td><p>CHARACTER STRING | Vertical axis label; default "Count"</p></td>
    </tr>
    <tr>
      <th>xAxisLabel</th>
      <td><p>CHARACTER STRING | Horizontal axis label; default 
"Sample"</p></td>
    </tr>
    <tr>
      <th>lineSize</th>
      <td><p>INTEGER | Size of plotted parallel coordinate lines; 
default 0.1</p></td>
    </tr>
    <tr>
      <th>lineAlpha</th>
      <td><p>INTEGER | Alpha value of plotted parallel coordinate 
lines, default 0.5</p></td>
    </tr>
    <tr>
      <th>vxAxis</th>
      <td><p>BOOLEAN [TRUE | FALSE] | Flip x-axis text labels to vertical
orientation; default FALSE</p></td>
    </tr>
    <tr>
      <th>outDir</th>
      <td><p>CHARACTER STRING | Output directory to save all images; 
default tempdir()</p></td>
    </tr>
    <tr>
      <th>saveFile</th>
      <td><p>BOOLEAN [TRUE | FALSE] | Save file to outDir; default TRUE</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>BOOLEAN [TRUE | FALSE] | Print each cluster from each 
cluster size into separate files and print the associated IDs of each 
cluster from each cluster size into separate .rds files; default is FALSE</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>List of n elements each containing a grid of parallel coordinate 
plots, where n is the number of treatment pair combinations in the data 
object. If the saveFile parameter has a value of TRUE, then each grid of 
parallel coordinate plots is saved to the location specified in the outDir
parameter as a JPG file. If the verbose parameter has a value of TRUE, 
then a JPG file for each parallel coordinate plot in each grid, RDS file 
containing the superimposed IDs for each parallel coordinate plot in each 
grid, and the JPG file of each grid of parallel coordinate plots is saved 
to the location specified in the outDir parameter.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='http://www.rdocumentation.org/packages/stats/topics/hclust'>hclust</a></code>
<a href='https://lindsayrutter.github.io/bigPint/articles/clusters.html'>https://lindsayrutter.github.io/bigPint/articles/clusters.html</a></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Example 1: Perform hierarchical clustering of size four using the </span>
<span class='co'># default agglomeration method "ward.D". Cluster only on the genes that have</span>
<span class='co'># FDR &lt; 1e-7 (n = 113) and overlay these genes.</span>

<span class='fu'>library</span>(<span class='no'>grid</span>)
<span class='fu'>library</span>(<span class='no'>matrixStats</span>)
<span class='fu'>library</span>(<span class='no'>ggplot2</span>)
<span class='fu'>data</span>(<span class='no'>soybean_ir_sub</span>)
<span class='no'>soybean_ir_sub</span>[,-<span class='fl'>1</span>] <span class='kw'>&lt;-</span> <span class='fu'>log</span>(<span class='no'>soybean_ir_sub</span>[-<span class='fl'>1</span>]+<span class='fl'>1</span>)
<span class='fu'>data</span>(<span class='no'>soybean_ir_sub_metrics</span>)
<span class='no'>colList</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"#00A600FF"</span>, <span class='fu'>rainbow</span>(<span class='fl'>5</span>)[<span class='fu'>c</span>(<span class='fl'>1</span>,<span class='fl'>4</span>,<span class='fl'>5</span>)])
<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotClusters</span>(<span class='kw'>data</span><span class='kw'>=</span><span class='no'>soybean_ir_sub</span>,
    <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='no'>soybean_ir_sub_metrics</span>, <span class='kw'>nC</span><span class='kw'>=</span><span class='fl'>4</span>, <span class='kw'>colList</span> <span class='kw'>=</span> <span class='no'>colList</span>,
    <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>threshVal</span> <span class='kw'>=</span> <span class='fl'>1e-7</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/grid/topics/grid.draw'>grid.draw</a></span>(<span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"N_P_4"</span>]])</div><div class='input'>
<span class='co'># Example 2: Perform the same analysis, only now create the four groups by </span>
<span class='co'># clustering on all genes in the data (n = 5,604). Then, overlay the genes </span>
<span class='co'># that have FDR &lt; 1e-7 (n = 113) into their corresponding clusters.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotClusters</span>(<span class='kw'>data</span><span class='kw'>=</span><span class='no'>soybean_ir_sub</span>,
    <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='no'>soybean_ir_sub_metrics</span>, <span class='kw'>nC</span><span class='kw'>=</span><span class='fl'>4</span>, <span class='kw'>colList</span> <span class='kw'>=</span> <span class='no'>colList</span>,
    <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>threshVal</span> <span class='kw'>=</span> <span class='fl'>1e-7</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/grid/topics/grid.draw'>grid.draw</a></span>(<span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"N_P_4"</span>]])</div><div class='input'>
<span class='co'># Example 3: Perform the same analysis, only now overlay all genes in the </span>
<span class='co'># data by keeping the dataMetrics object as its default value of NULL.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotClusters</span>(<span class='kw'>data</span><span class='kw'>=</span><span class='no'>soybean_ir_sub</span>, <span class='kw'>nC</span><span class='kw'>=</span><span class='fl'>4</span>, <span class='kw'>colList</span> <span class='kw'>=</span> <span class='no'>colList</span>,
    <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/grid/topics/grid.draw'>grid.draw</a></span>(<span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"N_P_4"</span>]])</div><div class='input'>
<span class='co'># Example 4: Visualization of gene clusters is usually performed on</span>
<span class='co'># standardized data. Here, hierarchical clustering of size four is performed</span>
<span class='co'># using the agglomeration method "average" on standardized data. Only genes </span>
<span class='co'># with FDR &lt; 0.05 are used for the clustering. Only two of the three </span>
<span class='co'># pairwise combinations of treatment groups (S1 and S2; S1 and S3) have any </span>
<span class='co'># genes with FDR &lt; 0.05. The output plots for these two pairs are examined. </span>

<span class='fu'>data</span>(<span class='no'>soybean_cn_sub</span>)
<span class='fu'>data</span>(<span class='no'>soybean_cn_sub_metrics</span>)
<span class='no'>soybean_cn_sub_st</span> <span class='kw'>&lt;-</span> <span class='fu'>as.data.frame</span>(<span class='fu'>t</span>(<span class='fu'>apply</span>(<span class='fu'>as.matrix</span>(<span class='no'>soybean_cn_sub</span>[,-<span class='fl'>1</span>]),
    <span class='fl'>1</span>, <span class='no'>scale</span>)))
<span class='no'>soybean_cn_sub_st</span>$<span class='no'>ID</span> <span class='kw'>&lt;-</span> <span class='fu'>as.character</span>(<span class='no'>soybean_cn_sub</span>$<span class='no'>ID</span>)
<span class='no'>soybean_cn_sub_st</span> <span class='kw'>&lt;-</span> <span class='no'>soybean_cn_sub_st</span>[,<span class='fu'>c</span>(<span class='fu'>length</span>(<span class='no'>soybean_cn_sub_st</span>),
    <span class='fl'>1</span>:<span class='fu'>length</span>(<span class='no'>soybean_cn_sub_st</span>)-<span class='fl'>1</span>)]
<span class='fu'>colnames</span>(<span class='no'>soybean_cn_sub_st</span>) <span class='kw'>&lt;-</span> <span class='fu'>colnames</span>(<span class='no'>soybean_cn_sub</span>)
<span class='no'>nID</span> <span class='kw'>&lt;-</span> <span class='fu'>which</span>(<span class='fu'>is.nan</span>(<span class='no'>soybean_cn_sub_st</span>[,<span class='fl'>2</span>]))
<span class='no'>soybean_cn_sub_st</span>[<span class='no'>nID</span>,<span class='fl'>2</span>:<span class='fu'>length</span>(<span class='no'>soybean_cn_sub_st</span>)] <span class='kw'>&lt;-</span> <span class='fl'>0</span>
<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotClusters</span>(<span class='kw'>data</span><span class='kw'>=</span><span class='no'>soybean_cn_sub_st</span>,
    <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='no'>soybean_cn_sub_metrics</span>, <span class='kw'>nC</span><span class='kw'>=</span><span class='fl'>4</span>,
    <span class='kw'>colList</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"#00A600FF"</span>, <span class='st'>"#CC00FFFF"</span>, <span class='st'>"red"</span>, <span class='st'>"darkorange"</span>),
    <span class='kw'>lineSize</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>lineAlpha</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
    <span class='kw'>aggMethod</span> <span class='kw'>=</span> <span class='st'>"average"</span>, <span class='kw'>yAxisLabel</span> <span class='kw'>=</span> <span class='st'>"Standardized read count"</span>,
    <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; [1] "S2_S3: There were no significant genes"</div><div class='input'><span class='fu'>names</span>(<span class='no'>ret</span>)</div><div class='output co'>#&gt; [1] "S1_S2_4" "S1_S3_4" "S2_S3_4"</div><div class='input'><span class='fu'><a href='http://www.rdocumentation.org/packages/grid/topics/grid.draw'>grid.draw</a></span>(<span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"S1_S2_4"</span>]])</div><div class='input'><span class='fu'><a href='http://www.rdocumentation.org/packages/grid/topics/grid.draw'>grid.draw</a></span>(<span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"S1_S3_4"</span>]])</div><div class='img'><img src='plotClusters-1.png' alt='' width='700' height='700' /></div><div class='input'>
# Example 5: Run the same analysis, only now set the verbose parameter to 
# value TRUE. This will save images of each individual cluster, .rds files 
# that contain the IDs within each cluster, and images of the conglomerate 
# clusters to outDir (default tempdir()).

</div><span class='co'># NOT RUN {</span>
<span class='fu'>plotClusters</span>(<span class='kw'>data</span><span class='kw'>=</span><span class='no'>soybean_cn_sub_st</span>, <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='no'>soybean_cn_sub_metrics</span>,
  <span class='kw'>nC</span><span class='kw'>=</span><span class='fl'>4</span>, <span class='kw'>colList</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"#00A600FF"</span>, <span class='st'>"#CC00FFFF"</span>, <span class='st'>"red"</span>, <span class='st'>"darkorange"</span>),
  <span class='kw'>lineSize</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>lineAlpha</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>clusterAllData</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>aggMethod</span> <span class='kw'>=</span> <span class='st'>"average"</span>, <span class='kw'>yAxisLabel</span> <span class='kw'>=</span> <span class='st'>"Standardized read count"</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='co'># }</span><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://github.com/lindsayrutter'>Lindsay Rutter</a>, <a href='https://github.com/dicook'>Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  
<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
<script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>


  </body>
</html>

