<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot static scatterplot matrices — plotSM • bigPint</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Plot static scatterplot matrices — plotSM" />

<meta property="og:description" content="Plot static scatterplot matrix. Optionally, superimpose
differentially expressed genes (DEGs) onto scatterplot matrix." />

<meta property="og:image" content="http://lindsayrutter.github.io/bigPint/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bigPint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Get Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/plotIntro.html">Introduction to bigPint plots</a>
    </li>
    <li>
      <a href="../articles/installation.html">Installation</a>
    </li>
    <li>
      <a href="../articles/data.html">Data object</a>
    </li>
    <li>
      <a href="../articles/dataMetrics.html">Data metrics object</a>
    </li>
    <li>
      <a href="../articles/createDataMetrics.html">Creating data metrics object</a>
    </li>
    <li>
      <a href="../articles/superimposeData.html">Alternatives to data metrics object</a>
    </li>
    <li>
      <a href="../articles/static.html">Producing static plots</a>
    </li>
    <li>
      <a href="../articles/interactive.html">Producing interactive plots</a>
    </li>
    <li>
      <a href="../articles/clusters.html">Hierarchical clustering</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Recommended RNA-seq pipeline</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    manuscripts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/manuscripts.html">Manuscripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/lindsayrutter/bigPint/">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot static scatterplot matrices</h1>
    <small class="dont-index">Source: <a href='https://github.com/lindsayrutter/bigPint/blob/master/R/plotSM.R'><code>R/plotSM.R</code></a></small>
    <div class="hidden name"><code>plotSM.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Plot static scatterplot matrix. Optionally, superimpose
differentially expressed genes (DEGs) onto scatterplot matrix.</p>
    
    </div>

    <pre class="usage"><span class='fu'>plotSM</span>(<span class='kw'>data</span> <span class='kw'>=</span> <span class='no'>data</span>, <span class='kw'>dataMetrics</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>geneList</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>threshVar</span> <span class='kw'>=</span> <span class='st'>"FDR"</span>, <span class='kw'>threshVal</span> <span class='kw'>=</span> <span class='fl'>0.05</span>, <span class='kw'>option</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"allPoints"</span>,
  <span class='st'>"foldChange"</span>, <span class='st'>"orthogonal"</span>, <span class='st'>"hexagon"</span>), <span class='kw'>xbins</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>threshFC</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>threshOrth</span> <span class='kw'>=</span> <span class='fl'>3</span>, <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='st'>"orange"</span>,
  <span class='kw'>outDir</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/tempfile'>tempdir</a></span>(), <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>DATA FRAME | Read counts</p></td>
    </tr>
    <tr>
      <th>dataMetrics</th>
      <td><p>LIST | Differential expression metrics; default NULL</p></td>
    </tr>
    <tr>
      <th>geneList</th>
      <td><p>CHARACTER ARRAY | List of gene IDs to be drawn onto the 
scatterplot matrix of all data. Use this parameter if you have predetermined 
genes to be drawn. Otherwise, use dataMetrics, threshVar, and threshVal to 
create genes to be drawn onto the scatterplot matrix; default NULL; used in 
"hexagon" and "allPoints"</p></td>
    </tr>
    <tr>
      <th>threshVar</th>
      <td><p>CHARACTER STRING | Name of column in dataMetrics object
that is used to threshold significance; default "FDR"; used in all options</p></td>
    </tr>
    <tr>
      <th>threshVal</th>
      <td><p>INTEGER | Maximum value to threshold significance from 
threshVar object; default 0.05; used in all options</p></td>
    </tr>
    <tr>
      <th>option</th>
      <td><p>CHARACTER STRING ["foldChange" | "orthogonal" | "hexagon" | 
"allPoints"] | The type of plot; default "allPoints"</p></td>
    </tr>
    <tr>
      <th>xbins</th>
      <td><p>INTEGER | Number of bins partitioning the range of the plot; 
default 10; used in option "hexagon"</p></td>
    </tr>
    <tr>
      <th>threshFC</th>
      <td><p>INTEGER | Threshold of fold change; default 3; used in
option "foldChange"</p></td>
    </tr>
    <tr>
      <th>threshOrth</th>
      <td><p>INTEGER | Threshold of orthogonal distance; default 3;
used in option "orthogonal"</p></td>
    </tr>
    <tr>
      <th>pointSize</th>
      <td><p>INTEGER | Size of plotted points; default 0.5; used for
DEGs in "hexagon" and "allPoints" and used for all points in "foldChange"
and "orthogonal"</p></td>
    </tr>
    <tr>
      <th>pointColor</th>
      <td><p>CHARACTER STRING | Color of overlaid points on scatterplot 
matrix; default "orange"; used for DEGs in "hexagon" and "allPoints" and
used for all points in "foldChange" and "orthogonal"</p></td>
    </tr>
    <tr>
      <th>outDir</th>
      <td><p>CHARACTER STRING | Output directory to save all plots; default 
tempdir(); used in all options</p></td>
    </tr>
    <tr>
      <th>saveFile</th>
      <td><p>BOOLEAN [TRUE | FALSE] | Save file to outDir; default TRUE; 
used in all options</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>List of n elements of scatterplot matrices, where n is the number of 
treatment pair combinations in the data object. The subset of genes that are 
superimposed are determined through the dataMetrics or geneList parameter.
If the saveFile parameter has a value of TRUE, then each of these
scatterplot matrices is saved to the location specified in the outDir
parameter as a JPG file.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>There are seven options:</p><ul>
<li><p>"foldChange": Plots DEGs onto scatterplot matrix of fold changes</p></li>
<li><p>"orthogonal": Plots DEGs onto scatterplot matrix of orthogonal 
distance</p></li>
<li><p>"hexagon": Plot DEGs onto scatterplot matrix of hexagon binning</p></li>
<li><p>"allPoints": Plot DEGs onto scatterplot matrix of all data points</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Read in data and metrics (need for all examples)</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>soybean_cn_sub</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>soybean_cn_sub_metrics</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>soybean_ir_sub</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>soybean_ir_sub_metrics</span>)

<span class='co'># Create standardized version of data (need for some examples)</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>matrixStats</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>ggplot2</span>)
<span class='no'>soybean_cn_sub_st</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/as.data.frame'>as.data.frame</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/t'>t</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/apply'>apply</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>as.matrix</a></span>(<span class='no'>soybean_cn_sub</span>[,-<span class='fl'>1</span>]),
    <span class='fl'>1</span>, <span class='no'>scale</span>)))
<span class='no'>soybean_cn_sub_st</span>$<span class='no'>ID</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/character'>as.character</a></span>(<span class='no'>soybean_cn_sub</span>$<span class='no'>ID</span>)
<span class='no'>soybean_cn_sub_st</span> <span class='kw'>&lt;-</span> <span class='no'>soybean_cn_sub_st</span>[,<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/length'>length</a></span>(<span class='no'>soybean_cn_sub_st</span>),
    <span class='fl'>1</span>:<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/length'>length</a></span>(<span class='no'>soybean_cn_sub_st</span>)-<span class='fl'>1</span>)]
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>colnames</a></span>(<span class='no'>soybean_cn_sub_st</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>colnames</a></span>(<span class='no'>soybean_cn_sub</span>)
<span class='no'>nID</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/which'>which</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/is.finite'>is.nan</a></span>(<span class='no'>soybean_cn_sub_st</span>[,<span class='fl'>2</span>]))
<span class='no'>soybean_cn_sub_st</span>[<span class='no'>nID</span>,<span class='fl'>2</span>:<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/length'>length</a></span>(<span class='no'>soybean_cn_sub_st</span>)] <span class='kw'>&lt;-</span> <span class='fl'>0</span>

<span class='co'># Example 1: Plot scatterplot matrix of points. Saves three plots to outDir </span>
<span class='co'># because saveFile equals TRUE by default.</span></div><span class='co'># NOT RUN {</span>
<span class='fu'>plotSM</span>(<span class='no'>soybean_cn_sub</span>, <span class='no'>soybean_cn_sub_metrics</span>)
<span class='co'># }</span><div class='input'>
<span class='co'># Example 2: Plot scatterplot matrix of points. Return list of plots so user </span>
<span class='co'># can tailor them (such as add title) and does not save to outDir because </span>
<span class='co'># saveFile equals FALSE.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_cn_sub</span>, <span class='no'>soybean_cn_sub_metrics</span>, <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='st'>"pink"</span>,
    <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='co'># Determine names of plots in returned list</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='no'>ret</span>)</div><div class='output co'>#&gt; [1] "S1_S2" "S1_S3" "S2_S3"</div><div class='input'><span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"S1_S2"</span>]] + <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span>(<span class='st'>"S1 versus S2"</span>)</div><div class='output co'>#&gt; <span class='warning'>Warning: Using `as.character()` on a quosure is deprecated as of rlang 0.3.0.</span>
#&gt; <span class='warning'>Please use `as_label()` or `as_name()` instead.</span>
#&gt; <span class='warning'><span style='color: #555555;'>This warning is displayed once per session.</span><span></span></div><div class='img'><img src='plotSM-1.png' alt='' width='700' height='700' /></div><div class='input'><span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"S1_S3"</span>]] + <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span>(<span class='st'>"S1 versus S3"</span>)</div><div class='img'><img src='plotSM-2.png' alt='' width='700' height='700' /></div><div class='input'><span class='no'>ret</span><span class='kw'>[[</span><span class='st'>"S2_S3"</span>]] + <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span>(<span class='st'>"S2 versus S3"</span>)</div><div class='img'><img src='plotSM-3.png' alt='' width='700' height='700' /></div><div class='input'>
<span class='co'># Example 3: Plot standardized data as scatterplot matrix of points.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_cn_sub_st</span>, <span class='no'>soybean_cn_sub_metrics</span>,
    <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='st'>"#00C379"</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ret</span><span class='kw'>[[</span><span class='fl'>1</span>]] + <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span>(<span class='st'>"Standardized read counts"</span>) +
<span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span>(<span class='st'>"Standardized read counts"</span>)</div><div class='img'><img src='plotSM-4.png' alt='' width='700' height='700' /></div><div class='input'>
<span class='co'># Example 4: Plot scatterplot matrix of hexagons.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_cn_sub</span>, <span class='no'>soybean_cn_sub_metrics</span>, <span class='kw'>option</span> <span class='kw'>=</span> <span class='st'>"hexagon"</span>,
    <span class='kw'>xbins</span> <span class='kw'>=</span> <span class='fl'>5</span>, <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ret</span><span class='kw'>[[</span><span class='fl'>2</span>]]</div><div class='img'><img src='plotSM-5.png' alt='' width='700' height='700' /></div><div class='input'>
<span class='co'># Example 5: Plot scatterplot matrix of orthogonal distance on the logged</span>
<span class='co'># data, first without considering the metrics dataset and then considering</span>
<span class='co'># it.</span>

<span class='no'>soybean_ir_sub</span>[,-<span class='fl'>1</span>] <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/Log'>log</a></span>(<span class='no'>soybean_ir_sub</span>[,-<span class='fl'>1</span>] + <span class='fl'>1</span>)
<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_ir_sub</span>, <span class='kw'>option</span> <span class='kw'>=</span> <span class='st'>"orthogonal"</span>, <span class='kw'>threshOrth</span> <span class='kw'>=</span> <span class='fl'>2.5</span>,
    <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>0.2</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ret</span><span class='kw'>[[</span><span class='fl'>1</span>]]</div><div class='img'><img src='plotSM-6.png' alt='' width='700' height='700' /></div><div class='input'><span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_ir_sub</span>, <span class='no'>soybean_ir_sub_metrics</span>, <span class='kw'>option</span> <span class='kw'>=</span> <span class='st'>"orthogonal"</span>,
    <span class='kw'>threshOrth</span> <span class='kw'>=</span> <span class='fl'>2.5</span>, <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>0.2</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ret</span><span class='kw'>[[</span><span class='fl'>1</span>]]</div><div class='img'><img src='plotSM-7.png' alt='' width='700' height='700' /></div><div class='input'>
<span class='co'># Example 6: Plot scatterplot matrix of fold change.</span>

<span class='no'>ret</span> <span class='kw'>&lt;-</span> <span class='fu'>plotSM</span>(<span class='no'>soybean_cn_sub</span>, <span class='no'>soybean_cn_sub_metrics</span>, <span class='kw'>option</span> <span class='kw'>=</span> <span class='st'>"foldChange"</span>,
    <span class='kw'>threshFC</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>0.2</span>, <span class='kw'>saveFile</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ret</span><span class='kw'>[[</span><span class='fl'>1</span>]]</div><div class='img'><img src='plotSM-8.png' alt='' width='700' height='700' /></div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://github.com/lindsayrutter'>Lindsay Rutter</a>, <a href='https://github.com/dicook'>Dianne Cook</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: 'API_KEY',
    indexName: 'INDEX_NAME',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>


  </body>
</html>

